{
 "cells": [
  {
   "cell_type": "code",
   "execution_count": 1,
   "id": "734ff25b-4091-44a6-b8de-1d8b5e93d0b9",
   "metadata": {},
   "outputs": [],
   "source": [
    "import pandas as pd\n",
    "import seaborn as sb"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 8,
   "id": "6b588320-1d8e-4061-a4db-b2754db57237",
   "metadata": {},
   "outputs": [],
   "source": [
    "raw = pd.read_csv(\"mukamel.txt\", sep='\\t')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 9,
   "id": "9d87ceeb-e328-4d74-b52d-36b0a4c34f18",
   "metadata": {},
   "outputs": [],
   "source": [
    "def get_coords(x):\n",
    "    data = x.split('_')\n",
    "    chrom, start, end = data[2:]\n",
    "    return chrom, start, end\n",
    "coords = raw['VNTR (hg38)'].apply(get_coords)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 10,
   "id": "94e8f1d8-8c69-45d6-83a1-75578343bbb9",
   "metadata": {},
   "outputs": [],
   "source": [
    "data = pd.concat([pd.DataFrame(list(coords), columns=['chrom', 'start', 'end']), raw], axis=1)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 11,
   "id": "64591b9f-dcc5-4d53-bb8a-a9caa9cee05a",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>chrom</th>\n",
       "      <th>start</th>\n",
       "      <th>end</th>\n",
       "      <th>VNTR (hg38)</th>\n",
       "      <th>VNTR (hg19)</th>\n",
       "      <th>Gene</th>\n",
       "      <th>Exon overlap pattern</th>\n",
       "      <th>Period (bp)</th>\n",
       "      <th>Repeats in hg38</th>\n",
       "      <th>IBD2 R(sib1,sib2)</th>\n",
       "      <th>Imputation R2 (est.)</th>\n",
       "      <th>Notes</th>\n",
       "      <th>Repeat unit</th>\n",
       "      <th>Ascertainment method</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>chr1</td>\n",
       "      <td>7829874</td>\n",
       "      <td>7830126</td>\n",
       "      <td>VNTR_HG38_chr1_7829874_7830126</td>\n",
       "      <td>VNTR_HG19_1_7889934_7890186</td>\n",
       "      <td>PER3</td>\n",
       "      <td>intra-exon</td>\n",
       "      <td>54</td>\n",
       "      <td>4.7</td>\n",
       "      <td>0.549</td>\n",
       "      <td>0.930</td>\n",
       "      <td>NaN</td>\n",
       "      <td>CTACTGCCAGCGCTCTGTCCACGGGATCGCCTCCCAGGAAGAATCC...</td>\n",
       "      <td>TRF</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>chr1</td>\n",
       "      <td>8358581</td>\n",
       "      <td>8358640</td>\n",
       "      <td>VNTR_HG38_chr1_8358581_8358640</td>\n",
       "      <td>VNTR_HG19_1_8418641_8418700</td>\n",
       "      <td>RERE</td>\n",
       "      <td>intra-exon</td>\n",
       "      <td>15</td>\n",
       "      <td>4.0</td>\n",
       "      <td>0.357</td>\n",
       "      <td>0.879</td>\n",
       "      <td>Upon detailed review, we believe the read-dept...</td>\n",
       "      <td>CTCCCGCTCTCGGAT</td>\n",
       "      <td>TRF</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>chr1</td>\n",
       "      <td>39413249</td>\n",
       "      <td>39413803</td>\n",
       "      <td>VNTR_HG38_chr1_39413249_39413803</td>\n",
       "      <td>VNTR_HG19_1_39878921_39879475</td>\n",
       "      <td>MACF1</td>\n",
       "      <td>intra-exon</td>\n",
       "      <td>39</td>\n",
       "      <td>14.2</td>\n",
       "      <td>0.303</td>\n",
       "      <td>0.562</td>\n",
       "      <td>NaN</td>\n",
       "      <td>CCACCCCAGAGGAGCCCACCTCCCCAGCTGCTGCAGTGC</td>\n",
       "      <td>TRF + VNTR scanner</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>chr1</td>\n",
       "      <td>62209956</td>\n",
       "      <td>62210059</td>\n",
       "      <td>VNTR_HG38_chr1_62209956_62210059</td>\n",
       "      <td>VNTR_HG19_1_62675628_62675731</td>\n",
       "      <td>L1TD1</td>\n",
       "      <td>intra-exon</td>\n",
       "      <td>33</td>\n",
       "      <td>3.2</td>\n",
       "      <td>0.515</td>\n",
       "      <td>0.901</td>\n",
       "      <td>NaN</td>\n",
       "      <td>GGAGGAGGAAGAAGAGACCTCAGGGCTGGAGGA</td>\n",
       "      <td>TRF</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>chr1</td>\n",
       "      <td>84574316</td>\n",
       "      <td>84574349</td>\n",
       "      <td>VNTR_HG38_chr1_84574316_84574349</td>\n",
       "      <td>VNTR_HG19_1_85039999_85040032</td>\n",
       "      <td>AL359762.1</td>\n",
       "      <td>intra-exon</td>\n",
       "      <td>9</td>\n",
       "      <td>3.8</td>\n",
       "      <td>0.392</td>\n",
       "      <td>0.909</td>\n",
       "      <td>NaN</td>\n",
       "      <td>GCAGCGCCA</td>\n",
       "      <td>TRF</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>113</th>\n",
       "      <td>chr21</td>\n",
       "      <td>44627465</td>\n",
       "      <td>44627515</td>\n",
       "      <td>VNTR_HG38_chr21_44627465_44627515</td>\n",
       "      <td>VNTR_HG19_21_46047382_46047432</td>\n",
       "      <td>KRTAP10-9</td>\n",
       "      <td>intra-exon</td>\n",
       "      <td>15</td>\n",
       "      <td>3.4</td>\n",
       "      <td>0.459</td>\n",
       "      <td>0.726</td>\n",
       "      <td>NaN</td>\n",
       "      <td>CTGCTGCGTGCCCGT</td>\n",
       "      <td>TRF</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>114</th>\n",
       "      <td>chr21</td>\n",
       "      <td>44637696</td>\n",
       "      <td>44637743</td>\n",
       "      <td>VNTR_HG38_chr21_44637696_44637743</td>\n",
       "      <td>VNTR_HG19_21_46057613_46057660</td>\n",
       "      <td>KRTAP10-10</td>\n",
       "      <td>intra-exon</td>\n",
       "      <td>15</td>\n",
       "      <td>3.2</td>\n",
       "      <td>0.330</td>\n",
       "      <td>1.312</td>\n",
       "      <td>NaN</td>\n",
       "      <td>CTGCTGTGTGCCCGT</td>\n",
       "      <td>TRF</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>115</th>\n",
       "      <td>chr21</td>\n",
       "      <td>46334453</td>\n",
       "      <td>46334765</td>\n",
       "      <td>VNTR_HG38_chr21_46334453_46334765</td>\n",
       "      <td>VNTR_HG19_21_47754367_47754679</td>\n",
       "      <td>PCNT</td>\n",
       "      <td>intra-exon</td>\n",
       "      <td>39</td>\n",
       "      <td>8.0</td>\n",
       "      <td>0.271</td>\n",
       "      <td>1.727</td>\n",
       "      <td>NaN</td>\n",
       "      <td>AGTCAGTGACCACCCACCAGAACAGCGTGGGATGTTCAC</td>\n",
       "      <td>TRF</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>116</th>\n",
       "      <td>chr22</td>\n",
       "      <td>37568317</td>\n",
       "      <td>37568479</td>\n",
       "      <td>VNTR_HG38_chr22_37568317_37568479</td>\n",
       "      <td>VNTR_HG19_22_37964324_37964486</td>\n",
       "      <td>CDC42EP1</td>\n",
       "      <td>intra-exon</td>\n",
       "      <td>21</td>\n",
       "      <td>7.8</td>\n",
       "      <td>0.380</td>\n",
       "      <td>1.267</td>\n",
       "      <td>NaN</td>\n",
       "      <td>CCAGCCCCTGCTGCAAACCCC</td>\n",
       "      <td>TRF</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>117</th>\n",
       "      <td>chr22</td>\n",
       "      <td>37723761</td>\n",
       "      <td>37724957</td>\n",
       "      <td>VNTR_HG38_chr22_37723761_37724957</td>\n",
       "      <td>VNTR_HG19_22_38119768_38120964</td>\n",
       "      <td>TRIOBP</td>\n",
       "      <td>intra-exon</td>\n",
       "      <td>69</td>\n",
       "      <td>17.3</td>\n",
       "      <td>0.877</td>\n",
       "      <td>0.901</td>\n",
       "      <td>NaN</td>\n",
       "      <td>CCAGAACATCCTGTGCCCAGCGGGACAATCCCAGAGCCTCCTCTCC...</td>\n",
       "      <td>TRF + VNTR scanner</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>118 rows × 14 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "     chrom     start       end                        VNTR (hg38)  \\\n",
       "0     chr1   7829874   7830126     VNTR_HG38_chr1_7829874_7830126   \n",
       "1     chr1   8358581   8358640     VNTR_HG38_chr1_8358581_8358640   \n",
       "2     chr1  39413249  39413803   VNTR_HG38_chr1_39413249_39413803   \n",
       "3     chr1  62209956  62210059   VNTR_HG38_chr1_62209956_62210059   \n",
       "4     chr1  84574316  84574349   VNTR_HG38_chr1_84574316_84574349   \n",
       "..     ...       ...       ...                                ...   \n",
       "113  chr21  44627465  44627515  VNTR_HG38_chr21_44627465_44627515   \n",
       "114  chr21  44637696  44637743  VNTR_HG38_chr21_44637696_44637743   \n",
       "115  chr21  46334453  46334765  VNTR_HG38_chr21_46334453_46334765   \n",
       "116  chr22  37568317  37568479  VNTR_HG38_chr22_37568317_37568479   \n",
       "117  chr22  37723761  37724957  VNTR_HG38_chr22_37723761_37724957   \n",
       "\n",
       "                        VNTR (hg19)        Gene Exon overlap pattern  \\\n",
       "0       VNTR_HG19_1_7889934_7890186        PER3           intra-exon   \n",
       "1       VNTR_HG19_1_8418641_8418700        RERE           intra-exon   \n",
       "2     VNTR_HG19_1_39878921_39879475       MACF1           intra-exon   \n",
       "3     VNTR_HG19_1_62675628_62675731       L1TD1           intra-exon   \n",
       "4     VNTR_HG19_1_85039999_85040032  AL359762.1           intra-exon   \n",
       "..                              ...         ...                  ...   \n",
       "113  VNTR_HG19_21_46047382_46047432   KRTAP10-9           intra-exon   \n",
       "114  VNTR_HG19_21_46057613_46057660  KRTAP10-10           intra-exon   \n",
       "115  VNTR_HG19_21_47754367_47754679        PCNT           intra-exon   \n",
       "116  VNTR_HG19_22_37964324_37964486    CDC42EP1           intra-exon   \n",
       "117  VNTR_HG19_22_38119768_38120964      TRIOBP           intra-exon   \n",
       "\n",
       "    Period (bp)  Repeats in hg38  IBD2 R(sib1,sib2)  Imputation R2 (est.)  \\\n",
       "0            54              4.7              0.549                 0.930   \n",
       "1            15              4.0              0.357                 0.879   \n",
       "2            39             14.2              0.303                 0.562   \n",
       "3            33              3.2              0.515                 0.901   \n",
       "4             9              3.8              0.392                 0.909   \n",
       "..          ...              ...                ...                   ...   \n",
       "113          15              3.4              0.459                 0.726   \n",
       "114          15              3.2              0.330                 1.312   \n",
       "115          39              8.0              0.271                 1.727   \n",
       "116          21              7.8              0.380                 1.267   \n",
       "117          69             17.3              0.877                 0.901   \n",
       "\n",
       "                                                 Notes  \\\n",
       "0                                                  NaN   \n",
       "1    Upon detailed review, we believe the read-dept...   \n",
       "2                                                  NaN   \n",
       "3                                                  NaN   \n",
       "4                                                  NaN   \n",
       "..                                                 ...   \n",
       "113                                                NaN   \n",
       "114                                                NaN   \n",
       "115                                                NaN   \n",
       "116                                                NaN   \n",
       "117                                                NaN   \n",
       "\n",
       "                                           Repeat unit Ascertainment method  \n",
       "0    CTACTGCCAGCGCTCTGTCCACGGGATCGCCTCCCAGGAAGAATCC...                  TRF  \n",
       "1                                      CTCCCGCTCTCGGAT                  TRF  \n",
       "2              CCACCCCAGAGGAGCCCACCTCCCCAGCTGCTGCAGTGC   TRF + VNTR scanner  \n",
       "3                    GGAGGAGGAAGAAGAGACCTCAGGGCTGGAGGA                  TRF  \n",
       "4                                            GCAGCGCCA                  TRF  \n",
       "..                                                 ...                  ...  \n",
       "113                                    CTGCTGCGTGCCCGT                  TRF  \n",
       "114                                    CTGCTGTGTGCCCGT                  TRF  \n",
       "115            AGTCAGTGACCACCCACCAGAACAGCGTGGGATGTTCAC                  TRF  \n",
       "116                              CCAGCCCCTGCTGCAAACCCC                  TRF  \n",
       "117  CCAGAACATCCTGTGCCCAGCGGGACAATCCCAGAGCCTCCTCTCC...   TRF + VNTR scanner  \n",
       "\n",
       "[118 rows x 14 columns]"
      ]
     },
     "execution_count": 11,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "data"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 13,
   "id": "1c564a88-c6f7-4bbb-bb72-014290145ddf",
   "metadata": {},
   "outputs": [],
   "source": [
    "data[['chrom', 'start', 'end', 'Gene', \"Repeat unit\", \"Exon overlap pattern\"]].to_csv(\"muk.bed\", sep='\\t', header=False, index=False)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 14,
   "id": "a363a054-048d-456f-ac6f-5cbc89b7b16c",
   "metadata": {},
   "outputs": [],
   "source": [
    "# chrom\tstart\tend\tLocus\tMotifs 1\tMotifs 2\tRepeat type\tRegion\tPath. repeats\tInheritance mode\n",
    "# And then I will add a column of mukamel where True == the pheno vntr set."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 15,
   "id": "065ead90-179b-4013-8c56-2838173d93eb",
   "metadata": {},
   "outputs": [],
   "source": [
    "patho = pd.read_csv(\"Patho.tsv\", sep='\\t')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 23,
   "id": "c104c98c-e798-4fe6-9810-f254f4bac880",
   "metadata": {},
   "outputs": [],
   "source": [
    "muka = data.rename(columns={'Gene': 'Locus', 'Repeat unit': \"Motifs 1\", \"Exon overlap pattern\":\"Region\"})\n",
    "muka[\"Motifs 2\"] = None\n",
    "muka[\"Path. repeats\"] = None\n",
    "muka[\"Disease\"] = None\n",
    "muka[\"Inheritance mode\"] = None\n",
    "muka[\"Repeat type\"] = \"VNTR\""
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 24,
   "id": "b745121f-aaeb-4285-8792-411447c9d837",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>chrom</th>\n",
       "      <th>start</th>\n",
       "      <th>end</th>\n",
       "      <th>Locus</th>\n",
       "      <th>Motifs 1</th>\n",
       "      <th>Motifs 2</th>\n",
       "      <th>Repeat type</th>\n",
       "      <th>Region</th>\n",
       "      <th>Path. repeats</th>\n",
       "      <th>Inheritance mode</th>\n",
       "      <th>Disease</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>chrX</td>\n",
       "      <td>148500631</td>\n",
       "      <td>148500691</td>\n",
       "      <td>AFF2</td>\n",
       "      <td>CCG</td>\n",
       "      <td>GCC</td>\n",
       "      <td>Standard</td>\n",
       "      <td>5'UTR</td>\n",
       "      <td>&gt;=200</td>\n",
       "      <td>X-linked recessive</td>\n",
       "      <td>FRAXE mental retardation (FRAXE)</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>chr2</td>\n",
       "      <td>100104799</td>\n",
       "      <td>100104824</td>\n",
       "      <td>AFF3</td>\n",
       "      <td>NaN</td>\n",
       "      <td>GCC</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Intron</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>chrX</td>\n",
       "      <td>67545316</td>\n",
       "      <td>67545385</td>\n",
       "      <td>AR</td>\n",
       "      <td>CAG</td>\n",
       "      <td>GCA</td>\n",
       "      <td>Standard</td>\n",
       "      <td>Coding</td>\n",
       "      <td>&gt;=40</td>\n",
       "      <td>X-linked recessive</td>\n",
       "      <td>Spinal and bulbar muscular atrophy (SBMA)</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>chrX</td>\n",
       "      <td>25013649</td>\n",
       "      <td>25013698</td>\n",
       "      <td>ARX_1</td>\n",
       "      <td>GCN</td>\n",
       "      <td>GCG</td>\n",
       "      <td>Imperfect GCN</td>\n",
       "      <td>Coding</td>\n",
       "      <td>&gt;=23,&gt;=18</td>\n",
       "      <td>X-linked recessive</td>\n",
       "      <td>Developmental and epileptic encephalopathy-1 (...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>chrX</td>\n",
       "      <td>25013529</td>\n",
       "      <td>25013565</td>\n",
       "      <td>ARX_2</td>\n",
       "      <td>GCN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Imperfect GCN</td>\n",
       "      <td>Coding</td>\n",
       "      <td>&gt;=20,&gt;=20,&gt;=23</td>\n",
       "      <td>X-linked recessive</td>\n",
       "      <td>Developmental and epileptic encephalopathy-1 (...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>113</th>\n",
       "      <td>chr21</td>\n",
       "      <td>44627465</td>\n",
       "      <td>44627515</td>\n",
       "      <td>KRTAP10-9</td>\n",
       "      <td>CTGCTGCGTGCCCGT</td>\n",
       "      <td>None</td>\n",
       "      <td>VNTR</td>\n",
       "      <td>intra-exon</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>114</th>\n",
       "      <td>chr21</td>\n",
       "      <td>44637696</td>\n",
       "      <td>44637743</td>\n",
       "      <td>KRTAP10-10</td>\n",
       "      <td>CTGCTGTGTGCCCGT</td>\n",
       "      <td>None</td>\n",
       "      <td>VNTR</td>\n",
       "      <td>intra-exon</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>115</th>\n",
       "      <td>chr21</td>\n",
       "      <td>46334453</td>\n",
       "      <td>46334765</td>\n",
       "      <td>PCNT</td>\n",
       "      <td>AGTCAGTGACCACCCACCAGAACAGCGTGGGATGTTCAC</td>\n",
       "      <td>None</td>\n",
       "      <td>VNTR</td>\n",
       "      <td>intra-exon</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>116</th>\n",
       "      <td>chr22</td>\n",
       "      <td>37568317</td>\n",
       "      <td>37568479</td>\n",
       "      <td>CDC42EP1</td>\n",
       "      <td>CCAGCCCCTGCTGCAAACCCC</td>\n",
       "      <td>None</td>\n",
       "      <td>VNTR</td>\n",
       "      <td>intra-exon</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>117</th>\n",
       "      <td>chr22</td>\n",
       "      <td>37723761</td>\n",
       "      <td>37724957</td>\n",
       "      <td>TRIOBP</td>\n",
       "      <td>CCAGAACATCCTGTGCCCAGCGGGACAATCCCAGAGCCTCCTCTCC...</td>\n",
       "      <td>None</td>\n",
       "      <td>VNTR</td>\n",
       "      <td>intra-exon</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>186 rows × 11 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "     chrom      start        end       Locus  \\\n",
       "0     chrX  148500631  148500691        AFF2   \n",
       "1     chr2  100104799  100104824        AFF3   \n",
       "2     chrX   67545316   67545385          AR   \n",
       "3     chrX   25013649   25013698       ARX_1   \n",
       "4     chrX   25013529   25013565       ARX_2   \n",
       "..     ...        ...        ...         ...   \n",
       "113  chr21   44627465   44627515   KRTAP10-9   \n",
       "114  chr21   44637696   44637743  KRTAP10-10   \n",
       "115  chr21   46334453   46334765        PCNT   \n",
       "116  chr22   37568317   37568479    CDC42EP1   \n",
       "117  chr22   37723761   37724957      TRIOBP   \n",
       "\n",
       "                                              Motifs 1 Motifs 2  \\\n",
       "0                                                  CCG      GCC   \n",
       "1                                                  NaN      GCC   \n",
       "2                                                  CAG      GCA   \n",
       "3                                                  GCN      GCG   \n",
       "4                                                  GCN      NaN   \n",
       "..                                                 ...      ...   \n",
       "113                                    CTGCTGCGTGCCCGT     None   \n",
       "114                                    CTGCTGTGTGCCCGT     None   \n",
       "115            AGTCAGTGACCACCCACCAGAACAGCGTGGGATGTTCAC     None   \n",
       "116                              CCAGCCCCTGCTGCAAACCCC     None   \n",
       "117  CCAGAACATCCTGTGCCCAGCGGGACAATCCCAGAGCCTCCTCTCC...     None   \n",
       "\n",
       "       Repeat type      Region   Path. repeats    Inheritance mode  \\\n",
       "0         Standard       5'UTR           >=200  X-linked recessive   \n",
       "1              NaN      Intron             NaN                 NaN   \n",
       "2         Standard      Coding            >=40  X-linked recessive   \n",
       "3    Imperfect GCN      Coding       >=23,>=18  X-linked recessive   \n",
       "4    Imperfect GCN      Coding  >=20,>=20,>=23  X-linked recessive   \n",
       "..             ...         ...             ...                 ...   \n",
       "113           VNTR  intra-exon            None                None   \n",
       "114           VNTR  intra-exon            None                None   \n",
       "115           VNTR  intra-exon            None                None   \n",
       "116           VNTR  intra-exon            None                None   \n",
       "117           VNTR  intra-exon            None                None   \n",
       "\n",
       "                                               Disease  \n",
       "0                     FRAXE mental retardation (FRAXE)  \n",
       "1                                                  NaN  \n",
       "2            Spinal and bulbar muscular atrophy (SBMA)  \n",
       "3    Developmental and epileptic encephalopathy-1 (...  \n",
       "4    Developmental and epileptic encephalopathy-1 (...  \n",
       "..                                                 ...  \n",
       "113                                               None  \n",
       "114                                               None  \n",
       "115                                               None  \n",
       "116                                               None  \n",
       "117                                               None  \n",
       "\n",
       "[186 rows x 11 columns]"
      ]
     },
     "execution_count": 24,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "full = pd.concat([patho, muka[patho.columns]])\n",
    "full"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 26,
   "id": "bfa52a7f-cedb-4e1a-86f2-b3d44ff0aa25",
   "metadata": {},
   "outputs": [],
   "source": [
    "full.sort_values(['chrom', 'start']).to_csv(\"new_patho.tsv\", sep='\\t', index=False)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "5c26b1d4-a202-4ef0-b0ba-cbab93ab807d",
   "metadata": {},
   "outputs": [],
   "source": []
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "Python 3 (ipykernel)",
   "language": "python",
   "name": "python3"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.9.6"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 5
}
