
TRGT:

- https://github.com/PacificBiosciences/trgt/

ONT Paper:
- https://www.science.org/doi/10.1126/sciadv.abm5386

	Comprehensive genetic diagnosis of tandem repeat expansion disorders with programmable targeted nanopore sequencing

	More than 50 neurological and neuromuscular diseases are caused by short tandem repeat (STR) expansions, with 37 different genes implicated to date. We describe the use of programmable targeted long-read sequencing with Oxford Nanoporeâ€™s ReadUntil function for parallel genotyping of all known neuropathogenic STRs in a single assay.


STRipy:
- https://www.stripy.org

	STRipy is a free and open source application created for genotyping short tandem repeats in the simplest way. It allows the genotyping of any STR locus in an aligned Illumina WGS/WES or targeted sequencing file. STRipy also detects whether an allele is in the pathogenic range for 58 diseases caused by 55 loci in 52 genes and provides the distributions of alleles in healthy populations (including the reference missing XYLT1 locus). Information acquired from the literature and in-house population analysis is available on STRipy's STRs database. Additionally, a separate version of STRipy (STRipy-pipeline) is available for a large scale analysis of STRs, supporting parallel processing and newest telomere-to-telomere reference genome. More information is available on the About page and the publication.

gnomAD:
- https://gnomad.broadinstitute.org/short-tandem-repeats?dataset=gnomad_r3


STRipy:
- https://www.stripy.org
Note this is largely a subset of gnomAD

	STRipy is a free and open source application created for genotyping short tandem repeats in the simplest way. It allows the genotyping of any STR locus in an aligned Illumina WGS/WES or targeted sequencing file. STRipy also detects whether an allele is in the pathogenic range for 58 diseases caused by 55 loci in 52 genes and provides the distributions of alleles in healthy populations (including the reference missing XYLT1 locus). Information acquired from the literature and in-house population analysis is available on STRipy's STRs database. Additionally, a separate version of STRipy (STRipy-pipeline) is available for a large scale analysis of STRs, supporting parallel processing and newest telomere-to-telomere reference genome. More information is available on the About page and the publication.


Consolidated through manual curation and placed in `Patho.tsv`

New Pathos that aren't inside of v1.1
FGF14: chr13:102161577-102161726

https://www.nejm.org/doi/full/10.1056/NEJMoa2207406



THAP11: chr16:67842864-67842950

https://pubmed.ncbi.nlm.nih.gov/37148549/

New VNTR phenotypic that are (mostly) inside of v1.1 from: https://www.ncbi.nlm.nih.gov/pmc/articles/PMC8549062/

