{
 "cells": [
  {
   "cell_type": "code",
   "execution_count": 1,
   "id": "b39befe5-9c92-4e00-8972-ad9f49cb32cd",
   "metadata": {},
   "outputs": [],
   "source": [
    "import re\n",
    "import requests\n",
    "from bs4 import BeautifulSoup\n",
    "\n",
    "URL = \"https://stripy.org/database\"\n",
    "\n",
    "page = requests.get(URL)\n",
    "\n",
    "soup = BeautifulSoup(page.content, \"html.parser\")"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 2,
   "id": "2688bcd1-6d83-47dd-b15d-ab1ed0096ca2",
   "metadata": {},
   "outputs": [],
   "source": [
    "import pandas as pd"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 36,
   "id": "db3bcf43-a476-427c-af5b-a3c01f54082e",
   "metadata": {},
   "outputs": [],
   "source": [
    "table = pd.read_html(URL)[0]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 37,
   "id": "db1836bb-4bb0-4701-9f4e-1bb83998919d",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Locus</th>\n",
       "      <th>Repeat type</th>\n",
       "      <th>Motif</th>\n",
       "      <th>Region</th>\n",
       "      <th>Path. repeats</th>\n",
       "      <th>Disease</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>AFF2</td>\n",
       "      <td>Standard</td>\n",
       "      <td>CCG</td>\n",
       "      <td>5â² UTR</td>\n",
       "      <td>&amp;GreaterEqual;200</td>\n",
       "      <td>FRAXE mental retardation (FRAXE)</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>AR</td>\n",
       "      <td>Standard</td>\n",
       "      <td>CAG</td>\n",
       "      <td>Coding</td>\n",
       "      <td>&amp;GreaterEqual;40</td>\n",
       "      <td>Spinal and bulbar muscular atrophy (SBMA)</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>ARX_1</td>\n",
       "      <td>Imperfect GCN</td>\n",
       "      <td>GCN</td>\n",
       "      <td>Coding</td>\n",
       "      <td>&amp;GreaterEqual;23&amp;GreaterEqual;18</td>\n",
       "      <td>Developmental and epileptic encephalopathy-1 (...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>ARX_2</td>\n",
       "      <td>Imperfect GCN</td>\n",
       "      <td>GCN</td>\n",
       "      <td>Coding</td>\n",
       "      <td>&amp;GreaterEqual;20&amp;GreaterEqual;20&amp;GreaterEqual;23</td>\n",
       "      <td>Developmental and epileptic encephalopathy-1 (...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>ATXN10</td>\n",
       "      <td>Standard</td>\n",
       "      <td>ATTCT</td>\n",
       "      <td>Intron</td>\n",
       "      <td>&amp;GreaterEqual;800</td>\n",
       "      <td>Spinocerebellar ataxia 10 (SCA10)</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>5</th>\n",
       "      <td>ATXN8OS</td>\n",
       "      <td>Standard</td>\n",
       "      <td>CTG</td>\n",
       "      <td>3â² UTR</td>\n",
       "      <td>&amp;GreaterEqual;80</td>\n",
       "      <td>Spinocerebellar ataxia 8 (SCA8)</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>6</th>\n",
       "      <td>ATN1</td>\n",
       "      <td>Standard</td>\n",
       "      <td>CAG</td>\n",
       "      <td>Coding</td>\n",
       "      <td>&amp;GreaterEqual;49</td>\n",
       "      <td>Dentatorubral-pallidoluysian atrophy (DRPLA)</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>7</th>\n",
       "      <td>ATXN1</td>\n",
       "      <td>Standard</td>\n",
       "      <td>CAG</td>\n",
       "      <td>Coding</td>\n",
       "      <td>&amp;GreaterEqual;39</td>\n",
       "      <td>Spinocerebellar ataxia 1 (SCA1)</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>8</th>\n",
       "      <td>ATXN2</td>\n",
       "      <td>Standard</td>\n",
       "      <td>CAG</td>\n",
       "      <td>Coding</td>\n",
       "      <td>&amp;GreaterEqual;35</td>\n",
       "      <td>Spinocerebellar ataxia 2 (SCA2)</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>9</th>\n",
       "      <td>ATXN3</td>\n",
       "      <td>Standard</td>\n",
       "      <td>CAG</td>\n",
       "      <td>Coding</td>\n",
       "      <td>&amp;GreaterEqual;56</td>\n",
       "      <td>Spinocerebellar ataxia 3 (SCA3)</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>10</th>\n",
       "      <td>ATXN7</td>\n",
       "      <td>Standard</td>\n",
       "      <td>CAG</td>\n",
       "      <td>Coding</td>\n",
       "      <td>&amp;GreaterEqual;36</td>\n",
       "      <td>Spinocerebellar ataxia 7 (SCA7)</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>11</th>\n",
       "      <td>BEAN1</td>\n",
       "      <td>Nested</td>\n",
       "      <td>TGGAA</td>\n",
       "      <td>Intron</td>\n",
       "      <td>&amp;GreaterEqual;500</td>\n",
       "      <td>Spinocerebellar ataxia 31 (SCA31)</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>12</th>\n",
       "      <td>C9ORF72</td>\n",
       "      <td>Standard</td>\n",
       "      <td>GGGGCC</td>\n",
       "      <td>Intron</td>\n",
       "      <td>&amp;GreaterEqual;24</td>\n",
       "      <td>Amyotrophic lateral sclerosis and/or frontotem...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>13</th>\n",
       "      <td>CACNA1A</td>\n",
       "      <td>Standard</td>\n",
       "      <td>CAG</td>\n",
       "      <td>Coding</td>\n",
       "      <td>&amp;GreaterEqual;21</td>\n",
       "      <td>Spinocerebellar ataxia 6 (SCA6)</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>14</th>\n",
       "      <td>CBL</td>\n",
       "      <td>Standard</td>\n",
       "      <td>CCG</td>\n",
       "      <td>5â² UTR</td>\n",
       "      <td>&amp;GreaterEqual;101</td>\n",
       "      <td>Jacobsen syndrome (JBS)</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>15</th>\n",
       "      <td>CNBP</td>\n",
       "      <td>Standard</td>\n",
       "      <td>CCTG</td>\n",
       "      <td>Intron</td>\n",
       "      <td>&amp;GreaterEqual;55</td>\n",
       "      <td>Myotonic dystrophy 2 (DM2)</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>16</th>\n",
       "      <td>COMP</td>\n",
       "      <td>Standard</td>\n",
       "      <td>GAC</td>\n",
       "      <td>Coding</td>\n",
       "      <td>&amp;GreaterEqual;6&amp;GreaterEqual;7</td>\n",
       "      <td>Multiple epiphyseal dysplasia (EDM1)Pseudoacho...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>17</th>\n",
       "      <td>DAB1</td>\n",
       "      <td>Nested</td>\n",
       "      <td>ATTTC</td>\n",
       "      <td>Intron</td>\n",
       "      <td>&amp;GreaterEqual;31</td>\n",
       "      <td>Spinocerebellar ataxia 37 (SCA37)</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>18</th>\n",
       "      <td>DIP2B</td>\n",
       "      <td>Standard</td>\n",
       "      <td>CGG</td>\n",
       "      <td>5â² UTR</td>\n",
       "      <td>&amp;GreaterEqual;273</td>\n",
       "      <td>Mental retardation, FRA12A type (MRFRA12A)</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>19</th>\n",
       "      <td>DMD</td>\n",
       "      <td>Standard</td>\n",
       "      <td>GAA</td>\n",
       "      <td>Intron</td>\n",
       "      <td>&amp;GreaterEqual;59</td>\n",
       "      <td>Duchenne muscular dystrophy (DMD)</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>20</th>\n",
       "      <td>DMPK</td>\n",
       "      <td>Standard</td>\n",
       "      <td>CTG</td>\n",
       "      <td>3â² UTR</td>\n",
       "      <td>&amp;GreaterEqual;50</td>\n",
       "      <td>Myotonic dystrophy 1 (DM1)</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>21</th>\n",
       "      <td>FMR1</td>\n",
       "      <td>Standard</td>\n",
       "      <td>CGG</td>\n",
       "      <td>5â² UTR</td>\n",
       "      <td>55-200&amp;GreaterEqual;201</td>\n",
       "      <td>Fragile X tremor/ataxia syndrome (FXTAS)Fragil...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>22</th>\n",
       "      <td>FOXL2</td>\n",
       "      <td>Imperfect GCN</td>\n",
       "      <td>GCN</td>\n",
       "      <td>Coding</td>\n",
       "      <td>&amp;GreaterEqual;19</td>\n",
       "      <td>Blepharophimosis, ptosis, and epicanthus inver...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>23</th>\n",
       "      <td>FXN</td>\n",
       "      <td>Standard</td>\n",
       "      <td>GAA</td>\n",
       "      <td>Intron</td>\n",
       "      <td>&amp;GreaterEqual;70</td>\n",
       "      <td>Friedreich ataxia (FRDA)</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>24</th>\n",
       "      <td>GIPC1</td>\n",
       "      <td>Standard</td>\n",
       "      <td>GGC</td>\n",
       "      <td>5â² UTR</td>\n",
       "      <td>&amp;GreaterEqual;73</td>\n",
       "      <td>Oculopharyngodistal myopathy 2 (OPDM2)</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>25</th>\n",
       "      <td>GLS</td>\n",
       "      <td>Standard</td>\n",
       "      <td>GCA</td>\n",
       "      <td>5â² UTR</td>\n",
       "      <td>&amp;GreaterEqual;680</td>\n",
       "      <td>Glutaminase deficiency with impaired intellect...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>26</th>\n",
       "      <td>HOXA13_1</td>\n",
       "      <td>Imperfect GCN</td>\n",
       "      <td>GCN</td>\n",
       "      <td>Coding</td>\n",
       "      <td>&amp;GreaterEqual;22</td>\n",
       "      <td>Hand-foot-genital syndrome (HFG)</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>27</th>\n",
       "      <td>HOXA13_2</td>\n",
       "      <td>Imperfect GCN</td>\n",
       "      <td>GCN</td>\n",
       "      <td>Coding</td>\n",
       "      <td>&amp;GreaterEqual;18</td>\n",
       "      <td>Hand-foot-genital syndrome (HFG)</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>28</th>\n",
       "      <td>HOXA13_3</td>\n",
       "      <td>Imperfect GCN</td>\n",
       "      <td>GCN</td>\n",
       "      <td>Coding</td>\n",
       "      <td>&amp;GreaterEqual;24</td>\n",
       "      <td>Hand-foot-genital syndrome (HFG)</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>29</th>\n",
       "      <td>HOXD13</td>\n",
       "      <td>Imperfect GCN</td>\n",
       "      <td>GCN</td>\n",
       "      <td>Coding</td>\n",
       "      <td>&amp;GreaterEqual;22</td>\n",
       "      <td>Synpolydactyly (SPD)</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>30</th>\n",
       "      <td>HTT</td>\n",
       "      <td>Standard</td>\n",
       "      <td>CAG</td>\n",
       "      <td>Coding</td>\n",
       "      <td>&amp;GreaterEqual;36</td>\n",
       "      <td>Huntington's disease (HD)</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>31</th>\n",
       "      <td>JPH3</td>\n",
       "      <td>Standard</td>\n",
       "      <td>CTG</td>\n",
       "      <td>3â² UTR</td>\n",
       "      <td>&amp;GreaterEqual;41</td>\n",
       "      <td>Huntington disease-like 2 (HDL2)</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>32</th>\n",
       "      <td>LRP12</td>\n",
       "      <td>Standard</td>\n",
       "      <td>CGG</td>\n",
       "      <td>5â² UTR</td>\n",
       "      <td>&amp;GreaterEqual;93</td>\n",
       "      <td>Oculopharyngodistal myopathy 1 (OPDM1)</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>33</th>\n",
       "      <td>MARCHF6</td>\n",
       "      <td>Nested</td>\n",
       "      <td>TTTCA</td>\n",
       "      <td>Intron</td>\n",
       "      <td>&amp;GreaterEqual;668</td>\n",
       "      <td>Familial adult myoclonic epilepsy 3 (FAME3)</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>34</th>\n",
       "      <td>NIPA1</td>\n",
       "      <td>Standard</td>\n",
       "      <td>GCG</td>\n",
       "      <td>5â² UTR</td>\n",
       "      <td>&amp;GreaterEqual;11</td>\n",
       "      <td>Amyotrophic lateral sclerosis (ALS)</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>35</th>\n",
       "      <td>NOTCH2NLC</td>\n",
       "      <td>Standard</td>\n",
       "      <td>GGC</td>\n",
       "      <td>5â² UTR</td>\n",
       "      <td>&amp;GreaterEqual;90&amp;GreaterEqual;60</td>\n",
       "      <td>Neuronal intranuclear inclusion disease (NIID)...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>36</th>\n",
       "      <td>NOP56</td>\n",
       "      <td>Standard</td>\n",
       "      <td>GGCCTG</td>\n",
       "      <td>Intron</td>\n",
       "      <td>&amp;GreaterEqual;25</td>\n",
       "      <td>Spinocerebellar ataxia 36 (SCA36)</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>37</th>\n",
       "      <td>NUTM2B-AS1</td>\n",
       "      <td>Standard</td>\n",
       "      <td>CGG</td>\n",
       "      <td>Noncoding transcript</td>\n",
       "      <td>&amp;GreaterEqual;700</td>\n",
       "      <td>Oculopharyngeal myopathy with leukoencephalopa...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>38</th>\n",
       "      <td>PABPN1</td>\n",
       "      <td>Standard</td>\n",
       "      <td>GCG</td>\n",
       "      <td>Coding</td>\n",
       "      <td>&amp;GreaterEqual;7</td>\n",
       "      <td>Oculopharyngeal muscular dystrophy (OPMD)</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>39</th>\n",
       "      <td>PHOX2B</td>\n",
       "      <td>Imperfect GCN</td>\n",
       "      <td>GCN</td>\n",
       "      <td>Coding</td>\n",
       "      <td>&amp;GreaterEqual;24</td>\n",
       "      <td>Central hypoventilation syndrome (CCHS)</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>40</th>\n",
       "      <td>PPP2R2B</td>\n",
       "      <td>Standard</td>\n",
       "      <td>CAG</td>\n",
       "      <td>5â² UTR</td>\n",
       "      <td>&amp;GreaterEqual;51</td>\n",
       "      <td>Spinocerebellar ataxia 12 (SCA12)</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>41</th>\n",
       "      <td>PRDM12</td>\n",
       "      <td>Standard</td>\n",
       "      <td>GCC</td>\n",
       "      <td>Coding</td>\n",
       "      <td>&amp;GreaterEqual;18</td>\n",
       "      <td>Hereditary sensory and autonomic neuropathy ty...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>42</th>\n",
       "      <td>RAPGEF2</td>\n",
       "      <td>Nested</td>\n",
       "      <td>TTTCA</td>\n",
       "      <td>Intron</td>\n",
       "      <td>&amp;GreaterEqual;60</td>\n",
       "      <td>Familial adult myoclonic epilepsy 7 (FAME7)</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>43</th>\n",
       "      <td>RFC1</td>\n",
       "      <td>Replaced</td>\n",
       "      <td>AAGGGACAGG</td>\n",
       "      <td>Intron</td>\n",
       "      <td>&amp;GreaterEqual;400</td>\n",
       "      <td>Cerebellar ataxia, neuropathy, and vestibular ...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>44</th>\n",
       "      <td>RUNX2</td>\n",
       "      <td>Imperfect GCN</td>\n",
       "      <td>GCN</td>\n",
       "      <td>Coding</td>\n",
       "      <td>&amp;GreaterEqual;27</td>\n",
       "      <td>Cleidocranial dysplasia (CCD)</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>45</th>\n",
       "      <td>SAMD12</td>\n",
       "      <td>Nested</td>\n",
       "      <td>TTTCA</td>\n",
       "      <td>Intron</td>\n",
       "      <td>&amp;GreaterEqual;100</td>\n",
       "      <td>Familial adult myoclonic epilepsy 1 (FAME1)</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>46</th>\n",
       "      <td>SOX3</td>\n",
       "      <td>Imperfect GCN</td>\n",
       "      <td>GCN</td>\n",
       "      <td>Coding</td>\n",
       "      <td>&amp;GreaterEqual;26&amp;GreaterEqual;22</td>\n",
       "      <td>X-linked mental retardation with isolated grow...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>47</th>\n",
       "      <td>STARD7</td>\n",
       "      <td>Nested</td>\n",
       "      <td>ATTTC</td>\n",
       "      <td>Intron</td>\n",
       "      <td>&amp;GreaterEqual;274</td>\n",
       "      <td>Familial adult myoclonic epilepsy 2 (FAME2)</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>48</th>\n",
       "      <td>TBP</td>\n",
       "      <td>Standard</td>\n",
       "      <td>CAG</td>\n",
       "      <td>Coding</td>\n",
       "      <td>&amp;GreaterEqual;43</td>\n",
       "      <td>Spinocerebellar ataxia 17 (SCA17)</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>49</th>\n",
       "      <td>TBX1</td>\n",
       "      <td>Imperfect GCN</td>\n",
       "      <td>GCN</td>\n",
       "      <td>Coding</td>\n",
       "      <td>&amp;GreaterEqual;25</td>\n",
       "      <td>Tetralogy of Fallot (TOF)</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>50</th>\n",
       "      <td>TCF4</td>\n",
       "      <td>Standard</td>\n",
       "      <td>CTG</td>\n",
       "      <td>Intron</td>\n",
       "      <td>&amp;GreaterEqual;50</td>\n",
       "      <td>Fuchs endothelial corneal dystrophy-3 (FECD3)</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>51</th>\n",
       "      <td>TNRC6A</td>\n",
       "      <td>Nested</td>\n",
       "      <td>TTTCA</td>\n",
       "      <td>Intron</td>\n",
       "      <td>&amp;GreaterEqual;1100</td>\n",
       "      <td>Familial adult myoclonic epilepsy 6 (FAME6)</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>52</th>\n",
       "      <td>XYLT1</td>\n",
       "      <td>Standard</td>\n",
       "      <td>GGC</td>\n",
       "      <td>5â² UTR</td>\n",
       "      <td>&amp;GreaterEqual;110</td>\n",
       "      <td>Desbuquois dysplasia-2 (DBQD2)</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>53</th>\n",
       "      <td>YEATS2</td>\n",
       "      <td>Nested</td>\n",
       "      <td>TTTCA</td>\n",
       "      <td>Intron</td>\n",
       "      <td>&amp;GreaterEqual;1000</td>\n",
       "      <td>Familial adult myoclonic epilepsy 4 (FAME4)</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>54</th>\n",
       "      <td>ZIC2</td>\n",
       "      <td>Imperfect GCN</td>\n",
       "      <td>GCN</td>\n",
       "      <td>Coding</td>\n",
       "      <td>&amp;GreaterEqual;25</td>\n",
       "      <td>Holoprosencephaly 5 (HPE5)</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>55</th>\n",
       "      <td>ZIC3</td>\n",
       "      <td>Standard</td>\n",
       "      <td>GCC</td>\n",
       "      <td>Coding</td>\n",
       "      <td>&amp;GreaterEqual;12</td>\n",
       "      <td>X-linked VACTERL syndrome (VACTERLX)</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "         Locus    Repeat type       Motif                Region  \\\n",
       "0         AFF2       Standard         CCG              5â² UTR   \n",
       "1           AR       Standard         CAG                Coding   \n",
       "2        ARX_1  Imperfect GCN         GCN                Coding   \n",
       "3        ARX_2  Imperfect GCN         GCN                Coding   \n",
       "4       ATXN10       Standard       ATTCT                Intron   \n",
       "5      ATXN8OS       Standard         CTG              3â² UTR   \n",
       "6         ATN1       Standard         CAG                Coding   \n",
       "7        ATXN1       Standard         CAG                Coding   \n",
       "8        ATXN2       Standard         CAG                Coding   \n",
       "9        ATXN3       Standard         CAG                Coding   \n",
       "10       ATXN7       Standard         CAG                Coding   \n",
       "11       BEAN1         Nested       TGGAA                Intron   \n",
       "12     C9ORF72       Standard      GGGGCC                Intron   \n",
       "13     CACNA1A       Standard         CAG                Coding   \n",
       "14         CBL       Standard         CCG              5â² UTR   \n",
       "15        CNBP       Standard        CCTG                Intron   \n",
       "16        COMP       Standard         GAC                Coding   \n",
       "17        DAB1         Nested       ATTTC                Intron   \n",
       "18       DIP2B       Standard         CGG              5â² UTR   \n",
       "19         DMD       Standard         GAA                Intron   \n",
       "20        DMPK       Standard         CTG              3â² UTR   \n",
       "21        FMR1       Standard         CGG              5â² UTR   \n",
       "22       FOXL2  Imperfect GCN         GCN                Coding   \n",
       "23         FXN       Standard         GAA                Intron   \n",
       "24       GIPC1       Standard         GGC              5â² UTR   \n",
       "25         GLS       Standard         GCA              5â² UTR   \n",
       "26    HOXA13_1  Imperfect GCN         GCN                Coding   \n",
       "27    HOXA13_2  Imperfect GCN         GCN                Coding   \n",
       "28    HOXA13_3  Imperfect GCN         GCN                Coding   \n",
       "29      HOXD13  Imperfect GCN         GCN                Coding   \n",
       "30         HTT       Standard         CAG                Coding   \n",
       "31        JPH3       Standard         CTG              3â² UTR   \n",
       "32       LRP12       Standard         CGG              5â² UTR   \n",
       "33     MARCHF6         Nested       TTTCA                Intron   \n",
       "34       NIPA1       Standard         GCG              5â² UTR   \n",
       "35   NOTCH2NLC       Standard         GGC              5â² UTR   \n",
       "36       NOP56       Standard      GGCCTG                Intron   \n",
       "37  NUTM2B-AS1       Standard         CGG  Noncoding transcript   \n",
       "38      PABPN1       Standard         GCG                Coding   \n",
       "39      PHOX2B  Imperfect GCN         GCN                Coding   \n",
       "40     PPP2R2B       Standard         CAG              5â² UTR   \n",
       "41      PRDM12       Standard         GCC                Coding   \n",
       "42     RAPGEF2         Nested       TTTCA                Intron   \n",
       "43        RFC1       Replaced  AAGGGACAGG                Intron   \n",
       "44       RUNX2  Imperfect GCN         GCN                Coding   \n",
       "45      SAMD12         Nested       TTTCA                Intron   \n",
       "46        SOX3  Imperfect GCN         GCN                Coding   \n",
       "47      STARD7         Nested       ATTTC                Intron   \n",
       "48         TBP       Standard         CAG                Coding   \n",
       "49        TBX1  Imperfect GCN         GCN                Coding   \n",
       "50        TCF4       Standard         CTG                Intron   \n",
       "51      TNRC6A         Nested       TTTCA                Intron   \n",
       "52       XYLT1       Standard         GGC              5â² UTR   \n",
       "53      YEATS2         Nested       TTTCA                Intron   \n",
       "54        ZIC2  Imperfect GCN         GCN                Coding   \n",
       "55        ZIC3       Standard         GCC                Coding   \n",
       "\n",
       "                                       Path. repeats  \\\n",
       "0                                  &GreaterEqual;200   \n",
       "1                                   &GreaterEqual;40   \n",
       "2                   &GreaterEqual;23&GreaterEqual;18   \n",
       "3   &GreaterEqual;20&GreaterEqual;20&GreaterEqual;23   \n",
       "4                                  &GreaterEqual;800   \n",
       "5                                   &GreaterEqual;80   \n",
       "6                                   &GreaterEqual;49   \n",
       "7                                   &GreaterEqual;39   \n",
       "8                                   &GreaterEqual;35   \n",
       "9                                   &GreaterEqual;56   \n",
       "10                                  &GreaterEqual;36   \n",
       "11                                 &GreaterEqual;500   \n",
       "12                                  &GreaterEqual;24   \n",
       "13                                  &GreaterEqual;21   \n",
       "14                                 &GreaterEqual;101   \n",
       "15                                  &GreaterEqual;55   \n",
       "16                    &GreaterEqual;6&GreaterEqual;7   \n",
       "17                                  &GreaterEqual;31   \n",
       "18                                 &GreaterEqual;273   \n",
       "19                                  &GreaterEqual;59   \n",
       "20                                  &GreaterEqual;50   \n",
       "21                           55-200&GreaterEqual;201   \n",
       "22                                  &GreaterEqual;19   \n",
       "23                                  &GreaterEqual;70   \n",
       "24                                  &GreaterEqual;73   \n",
       "25                                 &GreaterEqual;680   \n",
       "26                                  &GreaterEqual;22   \n",
       "27                                  &GreaterEqual;18   \n",
       "28                                  &GreaterEqual;24   \n",
       "29                                  &GreaterEqual;22   \n",
       "30                                  &GreaterEqual;36   \n",
       "31                                  &GreaterEqual;41   \n",
       "32                                  &GreaterEqual;93   \n",
       "33                                 &GreaterEqual;668   \n",
       "34                                  &GreaterEqual;11   \n",
       "35                  &GreaterEqual;90&GreaterEqual;60   \n",
       "36                                  &GreaterEqual;25   \n",
       "37                                 &GreaterEqual;700   \n",
       "38                                   &GreaterEqual;7   \n",
       "39                                  &GreaterEqual;24   \n",
       "40                                  &GreaterEqual;51   \n",
       "41                                  &GreaterEqual;18   \n",
       "42                                  &GreaterEqual;60   \n",
       "43                                 &GreaterEqual;400   \n",
       "44                                  &GreaterEqual;27   \n",
       "45                                 &GreaterEqual;100   \n",
       "46                  &GreaterEqual;26&GreaterEqual;22   \n",
       "47                                 &GreaterEqual;274   \n",
       "48                                  &GreaterEqual;43   \n",
       "49                                  &GreaterEqual;25   \n",
       "50                                  &GreaterEqual;50   \n",
       "51                                &GreaterEqual;1100   \n",
       "52                                 &GreaterEqual;110   \n",
       "53                                &GreaterEqual;1000   \n",
       "54                                  &GreaterEqual;25   \n",
       "55                                  &GreaterEqual;12   \n",
       "\n",
       "                                              Disease  \n",
       "0                    FRAXE mental retardation (FRAXE)  \n",
       "1           Spinal and bulbar muscular atrophy (SBMA)  \n",
       "2   Developmental and epileptic encephalopathy-1 (...  \n",
       "3   Developmental and epileptic encephalopathy-1 (...  \n",
       "4                   Spinocerebellar ataxia 10 (SCA10)  \n",
       "5                     Spinocerebellar ataxia 8 (SCA8)  \n",
       "6        Dentatorubral-pallidoluysian atrophy (DRPLA)  \n",
       "7                     Spinocerebellar ataxia 1 (SCA1)  \n",
       "8                     Spinocerebellar ataxia 2 (SCA2)  \n",
       "9                     Spinocerebellar ataxia 3 (SCA3)  \n",
       "10                    Spinocerebellar ataxia 7 (SCA7)  \n",
       "11                  Spinocerebellar ataxia 31 (SCA31)  \n",
       "12  Amyotrophic lateral sclerosis and/or frontotem...  \n",
       "13                    Spinocerebellar ataxia 6 (SCA6)  \n",
       "14                            Jacobsen syndrome (JBS)  \n",
       "15                         Myotonic dystrophy 2 (DM2)  \n",
       "16  Multiple epiphyseal dysplasia (EDM1)Pseudoacho...  \n",
       "17                  Spinocerebellar ataxia 37 (SCA37)  \n",
       "18         Mental retardation, FRA12A type (MRFRA12A)  \n",
       "19                  Duchenne muscular dystrophy (DMD)  \n",
       "20                         Myotonic dystrophy 1 (DM1)  \n",
       "21  Fragile X tremor/ataxia syndrome (FXTAS)Fragil...  \n",
       "22  Blepharophimosis, ptosis, and epicanthus inver...  \n",
       "23                           Friedreich ataxia (FRDA)  \n",
       "24             Oculopharyngodistal myopathy 2 (OPDM2)  \n",
       "25  Glutaminase deficiency with impaired intellect...  \n",
       "26                   Hand-foot-genital syndrome (HFG)  \n",
       "27                   Hand-foot-genital syndrome (HFG)  \n",
       "28                   Hand-foot-genital syndrome (HFG)  \n",
       "29                               Synpolydactyly (SPD)  \n",
       "30                          Huntington's disease (HD)  \n",
       "31                   Huntington disease-like 2 (HDL2)  \n",
       "32             Oculopharyngodistal myopathy 1 (OPDM1)  \n",
       "33        Familial adult myoclonic epilepsy 3 (FAME3)  \n",
       "34                Amyotrophic lateral sclerosis (ALS)  \n",
       "35  Neuronal intranuclear inclusion disease (NIID)...  \n",
       "36                  Spinocerebellar ataxia 36 (SCA36)  \n",
       "37  Oculopharyngeal myopathy with leukoencephalopa...  \n",
       "38          Oculopharyngeal muscular dystrophy (OPMD)  \n",
       "39            Central hypoventilation syndrome (CCHS)  \n",
       "40                  Spinocerebellar ataxia 12 (SCA12)  \n",
       "41  Hereditary sensory and autonomic neuropathy ty...  \n",
       "42        Familial adult myoclonic epilepsy 7 (FAME7)  \n",
       "43  Cerebellar ataxia, neuropathy, and vestibular ...  \n",
       "44                      Cleidocranial dysplasia (CCD)  \n",
       "45        Familial adult myoclonic epilepsy 1 (FAME1)  \n",
       "46  X-linked mental retardation with isolated grow...  \n",
       "47        Familial adult myoclonic epilepsy 2 (FAME2)  \n",
       "48                  Spinocerebellar ataxia 17 (SCA17)  \n",
       "49                          Tetralogy of Fallot (TOF)  \n",
       "50      Fuchs endothelial corneal dystrophy-3 (FECD3)  \n",
       "51        Familial adult myoclonic epilepsy 6 (FAME6)  \n",
       "52                     Desbuquois dysplasia-2 (DBQD2)  \n",
       "53        Familial adult myoclonic epilepsy 4 (FAME4)  \n",
       "54                         Holoprosencephaly 5 (HPE5)  \n",
       "55               X-linked VACTERL syndrome (VACTERLX)  "
      ]
     },
     "execution_count": 37,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "table"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 82,
   "id": "4da581d0-661a-4bbb-81f7-eb24c884cfe5",
   "metadata": {},
   "outputs": [],
   "source": [
    "rows = []\n",
    "for idx, i in table.iterrows():\n",
    "    n_url = URL + '/' + i[\"Locus\"]\n",
    "    d = pd.read_html(n_url)\n",
    "    d = d[0].set_index(0)\n",
    "    ref_region = d.loc[\"Reference region\"].iloc[0].strip().split(' ')[0]\n",
    "    ref_region = re.split(\":|-\", ref_region)\n",
    "    ref_region[1] = int(ref_region[1])\n",
    "    ref_region[2] = int(ref_region[2])\n",
    "    # Add repeat unit\n",
    "    #ref_region.append(d.loc[\"Repeat unit\"])\n",
    "    \n",
    "    rows.append(ref_region)\n",
    "    \n",
    "m_coords = pd.DataFrame(rows, columns=[\"chrom\", \"start\", \"end\"])"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 84,
   "id": "d3a3774b-c6fe-4d75-b7e1-3831ee3b3ac7",
   "metadata": {},
   "outputs": [],
   "source": [
    "table = table.join(m_coords)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 86,
   "id": "7c0db160-9023-4a32-8a04-1d271eb2cf75",
   "metadata": {},
   "outputs": [],
   "source": [
    "stripy = table"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 89,
   "id": "1e81bee2-cee2-4204-a7f7-301ab77cf03a",
   "metadata": {},
   "outputs": [],
   "source": [
    "patho = pd.read_csv(\"pathogenic_repeats.hg38.bed\", sep='\\t', header=None, names=[\"chrom\", \"start\", \"end\", \"meta\"])"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 91,
   "id": "0c80ec15-75fb-442d-b987-52bc579420c6",
   "metadata": {},
   "outputs": [],
   "source": [
    "rows = []\n",
    "for idx, i in patho.iterrows():\n",
    "    x = i['meta'].split(';')\n",
    "    Locus = x[0].split('=')[1]\n",
    "    motifs = x[1].split('=')[1].split(',')\n",
    "                       \n",
    "    rows.append([Locus, motifs])\n",
    "patho = patho.join(pd.DataFrame(rows, columns=[\"Locus\", \"Motifs\"]))"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 93,
   "id": "baa0f40f-3dd4-40d9-905c-ae1c124bbbc5",
   "metadata": {},
   "outputs": [],
   "source": [
    "patho.set_index(\"Locus\", inplace=True, drop=False)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 95,
   "id": "47f47997-05df-43e7-8b22-342df5a47bf3",
   "metadata": {},
   "outputs": [],
   "source": [
    "stripy.set_index(\"Locus\", inplace=True, drop=False)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 102,
   "id": "f5f95a6b-f044-4e08-97cd-22fc96193f76",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "10"
      ]
     },
     "execution_count": 102,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "(~patho.index.isin(stripy.index)).sum()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 104,
   "id": "1031fde4-2846-44aa-aaef-123be85cf2c6",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "10"
      ]
     },
     "execution_count": 104,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "(~stripy.index.isin(patho.index)).sum()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 120,
   "id": "6a46230d-62ef-4e11-b8d9-4f3c7fdd91ef",
   "metadata": {},
   "outputs": [],
   "source": [
    "patho[\"source\"] = \"patho\"\n",
    "stripy[\"source\"] = \"stripy\"\n",
    "full = pd.concat([patho, stripy], axis=0)\n",
    "full = full.sort_index()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 122,
   "id": "5aaea7e8-071c-4391-aadb-bde5bd884870",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "patho     56\n",
       "stripy    56\n",
       "Name: source, dtype: int64"
      ]
     },
     "execution_count": 122,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "full[\"source\"].value_counts()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 125,
   "id": "137c8951-dbc5-4946-ae32-35e614051e8f",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>chrom</th>\n",
       "      <th>start</th>\n",
       "      <th>end</th>\n",
       "      <th>source</th>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Locus</th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>AFF2</th>\n",
       "      <td>chrX</td>\n",
       "      <td>148500631</td>\n",
       "      <td>148500691</td>\n",
       "      <td>patho</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>AFF2</th>\n",
       "      <td>chrX</td>\n",
       "      <td>148500638</td>\n",
       "      <td>148500683</td>\n",
       "      <td>stripy</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>AFF3</th>\n",
       "      <td>chr2</td>\n",
       "      <td>100104799</td>\n",
       "      <td>100104824</td>\n",
       "      <td>patho</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>AR</th>\n",
       "      <td>chrX</td>\n",
       "      <td>67545316</td>\n",
       "      <td>67545385</td>\n",
       "      <td>patho</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>AR</th>\n",
       "      <td>chrX</td>\n",
       "      <td>67545316</td>\n",
       "      <td>67545385</td>\n",
       "      <td>stripy</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>YEATS2</th>\n",
       "      <td>chr3</td>\n",
       "      <td>183712187</td>\n",
       "      <td>183712222</td>\n",
       "      <td>stripy</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>ZIC2</th>\n",
       "      <td>chr13</td>\n",
       "      <td>99985448</td>\n",
       "      <td>99985494</td>\n",
       "      <td>patho</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>ZIC2</th>\n",
       "      <td>chr13</td>\n",
       "      <td>99985448</td>\n",
       "      <td>99985493</td>\n",
       "      <td>stripy</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>ZIC3</th>\n",
       "      <td>chrX</td>\n",
       "      <td>137566826</td>\n",
       "      <td>137566856</td>\n",
       "      <td>stripy</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>ZNF713</th>\n",
       "      <td>chr7</td>\n",
       "      <td>55887601</td>\n",
       "      <td>55887640</td>\n",
       "      <td>patho</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>112 rows × 4 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "        chrom      start        end  source\n",
       "Locus                                      \n",
       "AFF2     chrX  148500631  148500691   patho\n",
       "AFF2     chrX  148500638  148500683  stripy\n",
       "AFF3     chr2  100104799  100104824   patho\n",
       "AR       chrX   67545316   67545385   patho\n",
       "AR       chrX   67545316   67545385  stripy\n",
       "...       ...        ...        ...     ...\n",
       "YEATS2   chr3  183712187  183712222  stripy\n",
       "ZIC2    chr13   99985448   99985494   patho\n",
       "ZIC2    chr13   99985448   99985493  stripy\n",
       "ZIC3     chrX  137566826  137566856  stripy\n",
       "ZNF713   chr7   55887601   55887640   patho\n",
       "\n",
       "[112 rows x 4 columns]"
      ]
     },
     "execution_count": 125,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "full[[\"chrom\", \"start\", \"end\", \"source\"]]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 126,
   "id": "73ce47f3-a0b0-4b59-8427-4c5ee978f330",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "66"
      ]
     },
     "execution_count": 126,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "len(full[\"Locus\"].unique())"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 128,
   "id": "c7c395c2-d64e-4586-9a3a-6a54d9184a20",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>chrom</th>\n",
       "      <th>start</th>\n",
       "      <th>end</th>\n",
       "      <th>meta</th>\n",
       "      <th>Locus</th>\n",
       "      <th>Motifs</th>\n",
       "      <th>source</th>\n",
       "      <th>Repeat type</th>\n",
       "      <th>Motif</th>\n",
       "      <th>Region</th>\n",
       "      <th>Path. repeats</th>\n",
       "      <th>Disease</th>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Locus</th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>AFF2</th>\n",
       "      <td>chrX</td>\n",
       "      <td>148500631</td>\n",
       "      <td>148500691</td>\n",
       "      <td>ID=AFF2;MOTIFS=GCC;STRUC=(GCC)n</td>\n",
       "      <td>AFF2</td>\n",
       "      <td>[GCC]</td>\n",
       "      <td>patho</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>AFF2</th>\n",
       "      <td>chrX</td>\n",
       "      <td>148500638</td>\n",
       "      <td>148500683</td>\n",
       "      <td>NaN</td>\n",
       "      <td>AFF2</td>\n",
       "      <td>NaN</td>\n",
       "      <td>stripy</td>\n",
       "      <td>Standard</td>\n",
       "      <td>CCG</td>\n",
       "      <td>5â² UTR</td>\n",
       "      <td>&amp;GreaterEqual;200</td>\n",
       "      <td>FRAXE mental retardation (FRAXE)</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>AFF3</th>\n",
       "      <td>chr2</td>\n",
       "      <td>100104799</td>\n",
       "      <td>100104824</td>\n",
       "      <td>ID=AFF3;MOTIFS=GCC;STRUC=(GCC)n</td>\n",
       "      <td>AFF3</td>\n",
       "      <td>[GCC]</td>\n",
       "      <td>patho</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>AR</th>\n",
       "      <td>chrX</td>\n",
       "      <td>67545316</td>\n",
       "      <td>67545385</td>\n",
       "      <td>ID=AR;MOTIFS=GCA;STRUC=(GCA)n</td>\n",
       "      <td>AR</td>\n",
       "      <td>[GCA]</td>\n",
       "      <td>patho</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>AR</th>\n",
       "      <td>chrX</td>\n",
       "      <td>67545316</td>\n",
       "      <td>67545385</td>\n",
       "      <td>NaN</td>\n",
       "      <td>AR</td>\n",
       "      <td>NaN</td>\n",
       "      <td>stripy</td>\n",
       "      <td>Standard</td>\n",
       "      <td>CAG</td>\n",
       "      <td>Coding</td>\n",
       "      <td>&amp;GreaterEqual;40</td>\n",
       "      <td>Spinal and bulbar muscular atrophy (SBMA)</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>YEATS2</th>\n",
       "      <td>chr3</td>\n",
       "      <td>183712187</td>\n",
       "      <td>183712222</td>\n",
       "      <td>NaN</td>\n",
       "      <td>YEATS2</td>\n",
       "      <td>NaN</td>\n",
       "      <td>stripy</td>\n",
       "      <td>Nested</td>\n",
       "      <td>TTTCA</td>\n",
       "      <td>Intron</td>\n",
       "      <td>&amp;GreaterEqual;1000</td>\n",
       "      <td>Familial adult myoclonic epilepsy 4 (FAME4)</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>ZIC2</th>\n",
       "      <td>chr13</td>\n",
       "      <td>99985448</td>\n",
       "      <td>99985494</td>\n",
       "      <td>ID=ZIC2;MOTIFS=GCN;STRUC=(GCN)n</td>\n",
       "      <td>ZIC2</td>\n",
       "      <td>[GCN]</td>\n",
       "      <td>patho</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>ZIC2</th>\n",
       "      <td>chr13</td>\n",
       "      <td>99985448</td>\n",
       "      <td>99985493</td>\n",
       "      <td>NaN</td>\n",
       "      <td>ZIC2</td>\n",
       "      <td>NaN</td>\n",
       "      <td>stripy</td>\n",
       "      <td>Imperfect GCN</td>\n",
       "      <td>GCN</td>\n",
       "      <td>Coding</td>\n",
       "      <td>&amp;GreaterEqual;25</td>\n",
       "      <td>Holoprosencephaly 5 (HPE5)</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>ZIC3</th>\n",
       "      <td>chrX</td>\n",
       "      <td>137566826</td>\n",
       "      <td>137566856</td>\n",
       "      <td>NaN</td>\n",
       "      <td>ZIC3</td>\n",
       "      <td>NaN</td>\n",
       "      <td>stripy</td>\n",
       "      <td>Standard</td>\n",
       "      <td>GCC</td>\n",
       "      <td>Coding</td>\n",
       "      <td>&amp;GreaterEqual;12</td>\n",
       "      <td>X-linked VACTERL syndrome (VACTERLX)</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>ZNF713</th>\n",
       "      <td>chr7</td>\n",
       "      <td>55887601</td>\n",
       "      <td>55887640</td>\n",
       "      <td>ID=ZNF713;MOTIFS=CGG;STRUC=(CGG)n</td>\n",
       "      <td>ZNF713</td>\n",
       "      <td>[CGG]</td>\n",
       "      <td>patho</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>112 rows × 12 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "        chrom      start        end                               meta  \\\n",
       "Locus                                                                    \n",
       "AFF2     chrX  148500631  148500691    ID=AFF2;MOTIFS=GCC;STRUC=(GCC)n   \n",
       "AFF2     chrX  148500638  148500683                                NaN   \n",
       "AFF3     chr2  100104799  100104824    ID=AFF3;MOTIFS=GCC;STRUC=(GCC)n   \n",
       "AR       chrX   67545316   67545385      ID=AR;MOTIFS=GCA;STRUC=(GCA)n   \n",
       "AR       chrX   67545316   67545385                                NaN   \n",
       "...       ...        ...        ...                                ...   \n",
       "YEATS2   chr3  183712187  183712222                                NaN   \n",
       "ZIC2    chr13   99985448   99985494    ID=ZIC2;MOTIFS=GCN;STRUC=(GCN)n   \n",
       "ZIC2    chr13   99985448   99985493                                NaN   \n",
       "ZIC3     chrX  137566826  137566856                                NaN   \n",
       "ZNF713   chr7   55887601   55887640  ID=ZNF713;MOTIFS=CGG;STRUC=(CGG)n   \n",
       "\n",
       "         Locus Motifs  source    Repeat type  Motif    Region  \\\n",
       "Locus                                                           \n",
       "AFF2      AFF2  [GCC]   patho            NaN    NaN       NaN   \n",
       "AFF2      AFF2    NaN  stripy       Standard    CCG  5â² UTR   \n",
       "AFF3      AFF3  [GCC]   patho            NaN    NaN       NaN   \n",
       "AR          AR  [GCA]   patho            NaN    NaN       NaN   \n",
       "AR          AR    NaN  stripy       Standard    CAG    Coding   \n",
       "...        ...    ...     ...            ...    ...       ...   \n",
       "YEATS2  YEATS2    NaN  stripy         Nested  TTTCA    Intron   \n",
       "ZIC2      ZIC2  [GCN]   patho            NaN    NaN       NaN   \n",
       "ZIC2      ZIC2    NaN  stripy  Imperfect GCN    GCN    Coding   \n",
       "ZIC3      ZIC3    NaN  stripy       Standard    GCC    Coding   \n",
       "ZNF713  ZNF713  [CGG]   patho            NaN    NaN       NaN   \n",
       "\n",
       "             Path. repeats                                      Disease  \n",
       "Locus                                                                    \n",
       "AFF2                   NaN                                          NaN  \n",
       "AFF2     &GreaterEqual;200             FRAXE mental retardation (FRAXE)  \n",
       "AFF3                   NaN                                          NaN  \n",
       "AR                     NaN                                          NaN  \n",
       "AR        &GreaterEqual;40    Spinal and bulbar muscular atrophy (SBMA)  \n",
       "...                    ...                                          ...  \n",
       "YEATS2  &GreaterEqual;1000  Familial adult myoclonic epilepsy 4 (FAME4)  \n",
       "ZIC2                   NaN                                          NaN  \n",
       "ZIC2      &GreaterEqual;25                   Holoprosencephaly 5 (HPE5)  \n",
       "ZIC3      &GreaterEqual;12         X-linked VACTERL syndrome (VACTERLX)  \n",
       "ZNF713                 NaN                                          NaN  \n",
       "\n",
       "[112 rows x 12 columns]"
      ]
     },
     "execution_count": 128,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "x = full[\"Locus\"].value_counts()\n",
    "full.loc[]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 132,
   "id": "9a7b84e6-ef64-4dbe-afc3-929b5b5b81db",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>chrom</th>\n",
       "      <th>start</th>\n",
       "      <th>end</th>\n",
       "      <th>meta</th>\n",
       "      <th>Locus</th>\n",
       "      <th>Motifs</th>\n",
       "      <th>source</th>\n",
       "      <th>Repeat type</th>\n",
       "      <th>Motif</th>\n",
       "      <th>Region</th>\n",
       "      <th>Path. repeats</th>\n",
       "      <th>Disease</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>TMEM185A</th>\n",
       "      <td>chrX</td>\n",
       "      <td>149631763</td>\n",
       "      <td>149631782</td>\n",
       "      <td>ID=TMEM185A;MOTIFS=GCC;STRUC=(GCC)n</td>\n",
       "      <td>TMEM185A</td>\n",
       "      <td>[GCC]</td>\n",
       "      <td>patho</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>ZIC3</th>\n",
       "      <td>chrX</td>\n",
       "      <td>137566826</td>\n",
       "      <td>137566856</td>\n",
       "      <td>NaN</td>\n",
       "      <td>ZIC3</td>\n",
       "      <td>NaN</td>\n",
       "      <td>stripy</td>\n",
       "      <td>Standard</td>\n",
       "      <td>GCC</td>\n",
       "      <td>Coding</td>\n",
       "      <td>&amp;GreaterEqual;12</td>\n",
       "      <td>X-linked VACTERL syndrome (VACTERLX)</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>TBX1</th>\n",
       "      <td>chr22</td>\n",
       "      <td>19766762</td>\n",
       "      <td>19766807</td>\n",
       "      <td>NaN</td>\n",
       "      <td>TBX1</td>\n",
       "      <td>NaN</td>\n",
       "      <td>stripy</td>\n",
       "      <td>Imperfect GCN</td>\n",
       "      <td>GCN</td>\n",
       "      <td>Coding</td>\n",
       "      <td>&amp;GreaterEqual;25</td>\n",
       "      <td>Tetralogy of Fallot (TOF)</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>HOXA13_1</th>\n",
       "      <td>chr7</td>\n",
       "      <td>27199924</td>\n",
       "      <td>27199966</td>\n",
       "      <td>NaN</td>\n",
       "      <td>HOXA13_1</td>\n",
       "      <td>NaN</td>\n",
       "      <td>stripy</td>\n",
       "      <td>Imperfect GCN</td>\n",
       "      <td>GCN</td>\n",
       "      <td>Coding</td>\n",
       "      <td>&amp;GreaterEqual;22</td>\n",
       "      <td>Hand-foot-genital syndrome (HFG)</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>PRDM12</th>\n",
       "      <td>chr9</td>\n",
       "      <td>130681606</td>\n",
       "      <td>130681639</td>\n",
       "      <td>NaN</td>\n",
       "      <td>PRDM12</td>\n",
       "      <td>NaN</td>\n",
       "      <td>stripy</td>\n",
       "      <td>Standard</td>\n",
       "      <td>GCC</td>\n",
       "      <td>Coding</td>\n",
       "      <td>&amp;GreaterEqual;18</td>\n",
       "      <td>Hereditary sensory and autonomic neuropathy ty...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>NOTCH2NL</th>\n",
       "      <td>chr1</td>\n",
       "      <td>149390802</td>\n",
       "      <td>149390841</td>\n",
       "      <td>ID=NOTCH2NL;MOTIFS=GGC;STRUC=(GGC)n</td>\n",
       "      <td>NOTCH2NL</td>\n",
       "      <td>[GGC]</td>\n",
       "      <td>patho</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>HOXA13_3</th>\n",
       "      <td>chr7</td>\n",
       "      <td>27199678</td>\n",
       "      <td>27199732</td>\n",
       "      <td>NaN</td>\n",
       "      <td>HOXA13_3</td>\n",
       "      <td>NaN</td>\n",
       "      <td>stripy</td>\n",
       "      <td>Imperfect GCN</td>\n",
       "      <td>GCN</td>\n",
       "      <td>Coding</td>\n",
       "      <td>&amp;GreaterEqual;24</td>\n",
       "      <td>Hand-foot-genital syndrome (HFG)</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>HOXA13_2</th>\n",
       "      <td>chr7</td>\n",
       "      <td>27199825</td>\n",
       "      <td>27199861</td>\n",
       "      <td>NaN</td>\n",
       "      <td>HOXA13_2</td>\n",
       "      <td>NaN</td>\n",
       "      <td>stripy</td>\n",
       "      <td>Imperfect GCN</td>\n",
       "      <td>GCN</td>\n",
       "      <td>Coding</td>\n",
       "      <td>&amp;GreaterEqual;18</td>\n",
       "      <td>Hand-foot-genital syndrome (HFG)</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>AFF3</th>\n",
       "      <td>chr2</td>\n",
       "      <td>100104799</td>\n",
       "      <td>100104824</td>\n",
       "      <td>ID=AFF3;MOTIFS=GCC;STRUC=(GCC)n</td>\n",
       "      <td>AFF3</td>\n",
       "      <td>[GCC]</td>\n",
       "      <td>patho</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>HOXA13</th>\n",
       "      <td>chr7</td>\n",
       "      <td>27199825</td>\n",
       "      <td>27199862</td>\n",
       "      <td>ID=HOXA13;MOTIFS=NGC;STRUC=(NGC)n</td>\n",
       "      <td>HOXA13</td>\n",
       "      <td>[NGC]</td>\n",
       "      <td>patho</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>FRA10AC1</th>\n",
       "      <td>chr10</td>\n",
       "      <td>93702522</td>\n",
       "      <td>93702547</td>\n",
       "      <td>ID=FRA10AC1;MOTIFS=CCG;STRUC=(CCG)n</td>\n",
       "      <td>FRA10AC1</td>\n",
       "      <td>[CCG]</td>\n",
       "      <td>patho</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>DMD</th>\n",
       "      <td>chrX</td>\n",
       "      <td>31284557</td>\n",
       "      <td>31284605</td>\n",
       "      <td>NaN</td>\n",
       "      <td>DMD</td>\n",
       "      <td>NaN</td>\n",
       "      <td>stripy</td>\n",
       "      <td>Standard</td>\n",
       "      <td>GAA</td>\n",
       "      <td>Intron</td>\n",
       "      <td>&amp;GreaterEqual;59</td>\n",
       "      <td>Duchenne muscular dystrophy (DMD)</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>CSTB</th>\n",
       "      <td>chr21</td>\n",
       "      <td>43776443</td>\n",
       "      <td>43776479</td>\n",
       "      <td>ID=CSTB;MOTIFS=CGCGGGGCGGGG;STRUC=(CGCGGGGCGGGG)n</td>\n",
       "      <td>CSTB</td>\n",
       "      <td>[CGCGGGGCGGGG]</td>\n",
       "      <td>patho</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>C11ORF80</th>\n",
       "      <td>chr11</td>\n",
       "      <td>66744821</td>\n",
       "      <td>66744850</td>\n",
       "      <td>ID=C11ORF80;MOTIFS=GCG;STRUC=(GCG)n</td>\n",
       "      <td>C11ORF80</td>\n",
       "      <td>[GCG]</td>\n",
       "      <td>patho</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>ATXN8OS</th>\n",
       "      <td>chr13</td>\n",
       "      <td>70139383</td>\n",
       "      <td>70139428</td>\n",
       "      <td>NaN</td>\n",
       "      <td>ATXN8OS</td>\n",
       "      <td>NaN</td>\n",
       "      <td>stripy</td>\n",
       "      <td>Standard</td>\n",
       "      <td>CTG</td>\n",
       "      <td>3â² UTR</td>\n",
       "      <td>&amp;GreaterEqual;80</td>\n",
       "      <td>Spinocerebellar ataxia 8 (SCA8)</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>ATXN8</th>\n",
       "      <td>chr13</td>\n",
       "      <td>70139353</td>\n",
       "      <td>70139428</td>\n",
       "      <td>ID=ATXN8;MOTIFS=CTA,CTG;STRUC=(CTA)n(CTG)n</td>\n",
       "      <td>ATXN8</td>\n",
       "      <td>[CTA, CTG]</td>\n",
       "      <td>patho</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>ARX_2</th>\n",
       "      <td>chrX</td>\n",
       "      <td>25013529</td>\n",
       "      <td>25013565</td>\n",
       "      <td>NaN</td>\n",
       "      <td>ARX_2</td>\n",
       "      <td>NaN</td>\n",
       "      <td>stripy</td>\n",
       "      <td>Imperfect GCN</td>\n",
       "      <td>GCN</td>\n",
       "      <td>Coding</td>\n",
       "      <td>&amp;GreaterEqual;20&amp;GreaterEqual;20&amp;GreaterEqual;23</td>\n",
       "      <td>Developmental and epileptic encephalopathy-1 (...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>ARX_1</th>\n",
       "      <td>chrX</td>\n",
       "      <td>25013649</td>\n",
       "      <td>25013697</td>\n",
       "      <td>NaN</td>\n",
       "      <td>ARX_1</td>\n",
       "      <td>NaN</td>\n",
       "      <td>stripy</td>\n",
       "      <td>Imperfect GCN</td>\n",
       "      <td>GCN</td>\n",
       "      <td>Coding</td>\n",
       "      <td>&amp;GreaterEqual;23&amp;GreaterEqual;18</td>\n",
       "      <td>Developmental and epileptic encephalopathy-1 (...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>ARX</th>\n",
       "      <td>chrX</td>\n",
       "      <td>25013649</td>\n",
       "      <td>25013698</td>\n",
       "      <td>ID=ARX;MOTIFS=GCG;STRUC=(GCG)n</td>\n",
       "      <td>ARX</td>\n",
       "      <td>[GCG]</td>\n",
       "      <td>patho</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>ZNF713</th>\n",
       "      <td>chr7</td>\n",
       "      <td>55887601</td>\n",
       "      <td>55887640</td>\n",
       "      <td>ID=ZNF713;MOTIFS=CGG;STRUC=(CGG)n</td>\n",
       "      <td>ZNF713</td>\n",
       "      <td>[CGG]</td>\n",
       "      <td>patho</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "          chrom      start        end  \\\n",
       "TMEM185A   chrX  149631763  149631782   \n",
       "ZIC3       chrX  137566826  137566856   \n",
       "TBX1      chr22   19766762   19766807   \n",
       "HOXA13_1   chr7   27199924   27199966   \n",
       "PRDM12     chr9  130681606  130681639   \n",
       "NOTCH2NL   chr1  149390802  149390841   \n",
       "HOXA13_3   chr7   27199678   27199732   \n",
       "HOXA13_2   chr7   27199825   27199861   \n",
       "AFF3       chr2  100104799  100104824   \n",
       "HOXA13     chr7   27199825   27199862   \n",
       "FRA10AC1  chr10   93702522   93702547   \n",
       "DMD        chrX   31284557   31284605   \n",
       "CSTB      chr21   43776443   43776479   \n",
       "C11ORF80  chr11   66744821   66744850   \n",
       "ATXN8OS   chr13   70139383   70139428   \n",
       "ATXN8     chr13   70139353   70139428   \n",
       "ARX_2      chrX   25013529   25013565   \n",
       "ARX_1      chrX   25013649   25013697   \n",
       "ARX        chrX   25013649   25013698   \n",
       "ZNF713     chr7   55887601   55887640   \n",
       "\n",
       "                                                       meta     Locus  \\\n",
       "TMEM185A                ID=TMEM185A;MOTIFS=GCC;STRUC=(GCC)n  TMEM185A   \n",
       "ZIC3                                                    NaN      ZIC3   \n",
       "TBX1                                                    NaN      TBX1   \n",
       "HOXA13_1                                                NaN  HOXA13_1   \n",
       "PRDM12                                                  NaN    PRDM12   \n",
       "NOTCH2NL                ID=NOTCH2NL;MOTIFS=GGC;STRUC=(GGC)n  NOTCH2NL   \n",
       "HOXA13_3                                                NaN  HOXA13_3   \n",
       "HOXA13_2                                                NaN  HOXA13_2   \n",
       "AFF3                        ID=AFF3;MOTIFS=GCC;STRUC=(GCC)n      AFF3   \n",
       "HOXA13                    ID=HOXA13;MOTIFS=NGC;STRUC=(NGC)n    HOXA13   \n",
       "FRA10AC1                ID=FRA10AC1;MOTIFS=CCG;STRUC=(CCG)n  FRA10AC1   \n",
       "DMD                                                     NaN       DMD   \n",
       "CSTB      ID=CSTB;MOTIFS=CGCGGGGCGGGG;STRUC=(CGCGGGGCGGGG)n      CSTB   \n",
       "C11ORF80                ID=C11ORF80;MOTIFS=GCG;STRUC=(GCG)n  C11ORF80   \n",
       "ATXN8OS                                                 NaN   ATXN8OS   \n",
       "ATXN8            ID=ATXN8;MOTIFS=CTA,CTG;STRUC=(CTA)n(CTG)n     ATXN8   \n",
       "ARX_2                                                   NaN     ARX_2   \n",
       "ARX_1                                                   NaN     ARX_1   \n",
       "ARX                          ID=ARX;MOTIFS=GCG;STRUC=(GCG)n       ARX   \n",
       "ZNF713                    ID=ZNF713;MOTIFS=CGG;STRUC=(CGG)n    ZNF713   \n",
       "\n",
       "                  Motifs  source    Repeat type Motif    Region  \\\n",
       "TMEM185A           [GCC]   patho            NaN   NaN       NaN   \n",
       "ZIC3                 NaN  stripy       Standard   GCC    Coding   \n",
       "TBX1                 NaN  stripy  Imperfect GCN   GCN    Coding   \n",
       "HOXA13_1             NaN  stripy  Imperfect GCN   GCN    Coding   \n",
       "PRDM12               NaN  stripy       Standard   GCC    Coding   \n",
       "NOTCH2NL           [GGC]   patho            NaN   NaN       NaN   \n",
       "HOXA13_3             NaN  stripy  Imperfect GCN   GCN    Coding   \n",
       "HOXA13_2             NaN  stripy  Imperfect GCN   GCN    Coding   \n",
       "AFF3               [GCC]   patho            NaN   NaN       NaN   \n",
       "HOXA13             [NGC]   patho            NaN   NaN       NaN   \n",
       "FRA10AC1           [CCG]   patho            NaN   NaN       NaN   \n",
       "DMD                  NaN  stripy       Standard   GAA    Intron   \n",
       "CSTB      [CGCGGGGCGGGG]   patho            NaN   NaN       NaN   \n",
       "C11ORF80           [GCG]   patho            NaN   NaN       NaN   \n",
       "ATXN8OS              NaN  stripy       Standard   CTG  3â² UTR   \n",
       "ATXN8         [CTA, CTG]   patho            NaN   NaN       NaN   \n",
       "ARX_2                NaN  stripy  Imperfect GCN   GCN    Coding   \n",
       "ARX_1                NaN  stripy  Imperfect GCN   GCN    Coding   \n",
       "ARX                [GCG]   patho            NaN   NaN       NaN   \n",
       "ZNF713             [CGG]   patho            NaN   NaN       NaN   \n",
       "\n",
       "                                             Path. repeats  \\\n",
       "TMEM185A                                               NaN   \n",
       "ZIC3                                      &GreaterEqual;12   \n",
       "TBX1                                      &GreaterEqual;25   \n",
       "HOXA13_1                                  &GreaterEqual;22   \n",
       "PRDM12                                    &GreaterEqual;18   \n",
       "NOTCH2NL                                               NaN   \n",
       "HOXA13_3                                  &GreaterEqual;24   \n",
       "HOXA13_2                                  &GreaterEqual;18   \n",
       "AFF3                                                   NaN   \n",
       "HOXA13                                                 NaN   \n",
       "FRA10AC1                                               NaN   \n",
       "DMD                                       &GreaterEqual;59   \n",
       "CSTB                                                   NaN   \n",
       "C11ORF80                                               NaN   \n",
       "ATXN8OS                                   &GreaterEqual;80   \n",
       "ATXN8                                                  NaN   \n",
       "ARX_2     &GreaterEqual;20&GreaterEqual;20&GreaterEqual;23   \n",
       "ARX_1                     &GreaterEqual;23&GreaterEqual;18   \n",
       "ARX                                                    NaN   \n",
       "ZNF713                                                 NaN   \n",
       "\n",
       "                                                    Disease  \n",
       "TMEM185A                                                NaN  \n",
       "ZIC3                   X-linked VACTERL syndrome (VACTERLX)  \n",
       "TBX1                              Tetralogy of Fallot (TOF)  \n",
       "HOXA13_1                   Hand-foot-genital syndrome (HFG)  \n",
       "PRDM12    Hereditary sensory and autonomic neuropathy ty...  \n",
       "NOTCH2NL                                                NaN  \n",
       "HOXA13_3                   Hand-foot-genital syndrome (HFG)  \n",
       "HOXA13_2                   Hand-foot-genital syndrome (HFG)  \n",
       "AFF3                                                    NaN  \n",
       "HOXA13                                                  NaN  \n",
       "FRA10AC1                                                NaN  \n",
       "DMD                       Duchenne muscular dystrophy (DMD)  \n",
       "CSTB                                                    NaN  \n",
       "C11ORF80                                                NaN  \n",
       "ATXN8OS                     Spinocerebellar ataxia 8 (SCA8)  \n",
       "ATXN8                                                   NaN  \n",
       "ARX_2     Developmental and epileptic encephalopathy-1 (...  \n",
       "ARX_1     Developmental and epileptic encephalopathy-1 (...  \n",
       "ARX                                                     NaN  \n",
       "ZNF713                                                  NaN  "
      ]
     },
     "execution_count": 132,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "x = full[\"Locus\"].value_counts() \n",
    "full.loc[x[x != 2].index]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 133,
   "id": "9a9eefc8-5b7b-451d-b31c-8e4a04219497",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "patho     10\n",
       "stripy    10\n",
       "Name: source, dtype: int64"
      ]
     },
     "execution_count": 133,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "full.loc[x[x != 2].index][\"source\"].value_counts()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 134,
   "id": "8e9f90de-59b0-4f9e-b549-7361cf909c7a",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "(56, 56)"
      ]
     },
     "execution_count": 134,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "len(patho), len(stripy)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 15,
   "id": "06c82455-07ab-4669-93ac-f95b091cd944",
   "metadata": {},
   "outputs": [],
   "source": [
    "patho = pd.read_csv(\"Patho.tsv\", sep='\\t')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 16,
   "id": "f459b925-7e3a-4854-8ae7-7c79fe1063e6",
   "metadata": {},
   "outputs": [],
   "source": [
    "gnom = pd.read_csv(\"gnomad_roughtable.txt\", sep='\\t')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 17,
   "id": "2c549794-6049-4fac-a796-982d9f89d94b",
   "metadata": {},
   "outputs": [],
   "source": [
    "gnom.set_index(\"ID\", inplace=True, drop=False)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 18,
   "id": "05d1fda0-a843-473c-927e-b83bcecd0f7c",
   "metadata": {},
   "outputs": [],
   "source": [
    "patho.set_index(\"Locus\", inplace=True, drop=False)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 19,
   "id": "e52253af-6526-4f2f-b4de-5f529616d5d1",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "58"
      ]
     },
     "execution_count": 19,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "patho.index.isin(gnom.index).sum()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 20,
   "id": "431222d1-ae64-4ce0-812d-963b3af6349e",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "58"
      ]
     },
     "execution_count": 20,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "gnom.index.isin(patho.index).sum()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 21,
   "id": "fda3c58e-162d-434e-945b-473c7e7d5167",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>ID</th>\n",
       "      <th>Reference repeat unit</th>\n",
       "      <th>Region</th>\n",
       "      <th>Inheritance mode</th>\n",
       "      <th>Associated disease(s)</th>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>ID</th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>ATXN8OS</th>\n",
       "      <td>ATXN8OS</td>\n",
       "      <td>CTG</td>\n",
       "      <td>3'-UTR</td>\n",
       "      <td>Autosomal dominant</td>\n",
       "      <td>Spinocerebellar ataxia 8</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "              ID Reference repeat unit  Region    Inheritance mode  \\\n",
       "ID                                                                   \n",
       "ATXN8OS  ATXN8OS                   CTG  3'-UTR  Autosomal dominant   \n",
       "\n",
       "            Associated disease(s)  \n",
       "ID                                 \n",
       "ATXN8OS  Spinocerebellar ataxia 8  "
      ]
     },
     "execution_count": 21,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "gnom[~gnom.index.isin(patho.index)]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 22,
   "id": "c57973ed-b386-47a4-9385-5d1e9c5197f0",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "66"
      ]
     },
     "execution_count": 22,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "len(patho)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 24,
   "id": "9f4a785f-8d7d-47e6-9b28-1fc3a7573fda",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>chrom</th>\n",
       "      <th>start</th>\n",
       "      <th>end</th>\n",
       "      <th>Locus</th>\n",
       "      <th>Motifs</th>\n",
       "      <th>source</th>\n",
       "      <th>Repeat type</th>\n",
       "      <th>Motif</th>\n",
       "      <th>Region</th>\n",
       "      <th>Path. repeats</th>\n",
       "      <th>Disease</th>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Locus</th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>AFF2</th>\n",
       "      <td>chrX</td>\n",
       "      <td>148500631</td>\n",
       "      <td>148500691</td>\n",
       "      <td>AFF2</td>\n",
       "      <td>GCC</td>\n",
       "      <td>both</td>\n",
       "      <td>Standard</td>\n",
       "      <td>CCG</td>\n",
       "      <td>5'UTR</td>\n",
       "      <td>&gt;=200</td>\n",
       "      <td>FRAXE mental retardation (FRAXE)</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>AFF3</th>\n",
       "      <td>chr2</td>\n",
       "      <td>100104799</td>\n",
       "      <td>100104824</td>\n",
       "      <td>AFF3</td>\n",
       "      <td>GCC</td>\n",
       "      <td>patho</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>AR</th>\n",
       "      <td>chrX</td>\n",
       "      <td>67545316</td>\n",
       "      <td>67545385</td>\n",
       "      <td>AR</td>\n",
       "      <td>GCA</td>\n",
       "      <td>both</td>\n",
       "      <td>Standard</td>\n",
       "      <td>CAG</td>\n",
       "      <td>Coding</td>\n",
       "      <td>&gt;=40</td>\n",
       "      <td>Spinal and bulbar muscular atrophy (SBMA)</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>ARX_1</th>\n",
       "      <td>chrX</td>\n",
       "      <td>25013649</td>\n",
       "      <td>25013698</td>\n",
       "      <td>ARX_1</td>\n",
       "      <td>GCG</td>\n",
       "      <td>both</td>\n",
       "      <td>Imperfect GCN</td>\n",
       "      <td>GCN</td>\n",
       "      <td>Coding</td>\n",
       "      <td>&gt;=23&gt;=18</td>\n",
       "      <td>Developmental and epileptic encephalopathy-1 (...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>ARX_2</th>\n",
       "      <td>chrX</td>\n",
       "      <td>25013529</td>\n",
       "      <td>25013565</td>\n",
       "      <td>ARX_2</td>\n",
       "      <td>NaN</td>\n",
       "      <td>stripy</td>\n",
       "      <td>Imperfect GCN</td>\n",
       "      <td>GCN</td>\n",
       "      <td>Coding</td>\n",
       "      <td>&gt;=20&gt;=20&gt;=23</td>\n",
       "      <td>Developmental and epileptic encephalopathy-1 (...</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "      chrom      start        end  Locus Motifs  source    Repeat type Motif  \\\n",
       "Locus                                                                          \n",
       "AFF2   chrX  148500631  148500691   AFF2    GCC    both       Standard   CCG   \n",
       "AFF3   chr2  100104799  100104824   AFF3    GCC   patho            NaN   NaN   \n",
       "AR     chrX   67545316   67545385     AR    GCA    both       Standard   CAG   \n",
       "ARX_1  chrX   25013649   25013698  ARX_1    GCG    both  Imperfect GCN   GCN   \n",
       "ARX_2  chrX   25013529   25013565  ARX_2    NaN  stripy  Imperfect GCN   GCN   \n",
       "\n",
       "       Region Path. repeats                                            Disease  \n",
       "Locus                                                                           \n",
       "AFF2    5'UTR         >=200                   FRAXE mental retardation (FRAXE)  \n",
       "AFF3      NaN           NaN                                                NaN  \n",
       "AR     Coding          >=40          Spinal and bulbar muscular atrophy (SBMA)  \n",
       "ARX_1  Coding      >=23>=18  Developmental and epileptic encephalopathy-1 (...  \n",
       "ARX_2  Coding  >=20>=20>=23  Developmental and epileptic encephalopathy-1 (...  "
      ]
     },
     "execution_count": 24,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "patho.head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 25,
   "id": "5c206773-05ed-453d-b14b-ae197d88e38f",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>ID</th>\n",
       "      <th>Reference repeat unit</th>\n",
       "      <th>Region</th>\n",
       "      <th>Inheritance mode</th>\n",
       "      <th>Associated disease(s)</th>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>ID</th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>AFF2</th>\n",
       "      <td>AFF2</td>\n",
       "      <td>GCC</td>\n",
       "      <td>5'-UTR</td>\n",
       "      <td>X-linked recessive</td>\n",
       "      <td>FRAXE mental retardation</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>AR</th>\n",
       "      <td>AR</td>\n",
       "      <td>GCA</td>\n",
       "      <td>coding: polyglutamine</td>\n",
       "      <td>X-linked recessive</td>\n",
       "      <td>Spinal and bulbar muscular atrophy</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>ARX_1</th>\n",
       "      <td>ARX_1</td>\n",
       "      <td>NGC</td>\n",
       "      <td>coding: polyalanine</td>\n",
       "      <td>X-linked recessive</td>\n",
       "      <td>Developmental and epileptic encephalopathy-1, ...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>ARX_2</th>\n",
       "      <td>ARX_2</td>\n",
       "      <td>NGC</td>\n",
       "      <td>coding: polyalanine</td>\n",
       "      <td>X-linked recessive</td>\n",
       "      <td>Developmental and epileptic encephalopathy-1, ...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>ATN1</th>\n",
       "      <td>ATN1</td>\n",
       "      <td>CAG</td>\n",
       "      <td>coding: polyglutamine</td>\n",
       "      <td>Autosomal dominant</td>\n",
       "      <td>Dentatorubral-pallidoluysian atrophy</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "          ID Reference repeat unit                 Region    Inheritance mode  \\\n",
       "ID                                                                              \n",
       "AFF2    AFF2                   GCC                 5'-UTR  X-linked recessive   \n",
       "AR        AR                   GCA  coding: polyglutamine  X-linked recessive   \n",
       "ARX_1  ARX_1                   NGC    coding: polyalanine  X-linked recessive   \n",
       "ARX_2  ARX_2                   NGC    coding: polyalanine  X-linked recessive   \n",
       "ATN1    ATN1                   CAG  coding: polyglutamine  Autosomal dominant   \n",
       "\n",
       "                                   Associated disease(s)  \n",
       "ID                                                        \n",
       "AFF2                            FRAXE mental retardation  \n",
       "AR                    Spinal and bulbar muscular atrophy  \n",
       "ARX_1  Developmental and epileptic encephalopathy-1, ...  \n",
       "ARX_2  Developmental and epileptic encephalopathy-1, ...  \n",
       "ATN1                Dentatorubral-pallidoluysian atrophy  "
      ]
     },
     "execution_count": 25,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "gnom.head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 27,
   "id": "52173329-ce1d-4064-b44c-93cf582185f8",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>ID</th>\n",
       "      <th>Reference repeat unit</th>\n",
       "      <th>Region</th>\n",
       "      <th>Inheritance mode</th>\n",
       "      <th>Associated disease(s)</th>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Locus</th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>PRNP</th>\n",
       "      <td>PRNP</td>\n",
       "      <td>CCTCATGGTGGTGGCTGGGGGCAG</td>\n",
       "      <td>coding: other</td>\n",
       "      <td>Autosomal dominant</td>\n",
       "      <td>Creutzfeldt-Jakob disease</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>VWA1</th>\n",
       "      <td>VWA1</td>\n",
       "      <td>GGCGCGGAGC</td>\n",
       "      <td>coding: other</td>\n",
       "      <td>Autosomal recessive</td>\n",
       "      <td>Hereditary motor neuropathy</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>EIF4A3</th>\n",
       "      <td>EIF4A3</td>\n",
       "      <td>CCTCGCTGTGCCGCTGCCGA</td>\n",
       "      <td>5'-UTR</td>\n",
       "      <td>Autosomal recessive</td>\n",
       "      <td>Richieri-Costa-Pereira syndrome</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>NIPA1</th>\n",
       "      <td>NIPA1</td>\n",
       "      <td>GCG</td>\n",
       "      <td>coding: polyalanine</td>\n",
       "      <td>Autosomal dominant</td>\n",
       "      <td>Amyotrophic lateral sclerosis</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "            ID     Reference repeat unit               Region  \\\n",
       "Locus                                                           \n",
       "PRNP      PRNP  CCTCATGGTGGTGGCTGGGGGCAG        coding: other   \n",
       "VWA1      VWA1                GGCGCGGAGC        coding: other   \n",
       "EIF4A3  EIF4A3      CCTCGCTGTGCCGCTGCCGA               5'-UTR   \n",
       "NIPA1    NIPA1                       GCG  coding: polyalanine   \n",
       "\n",
       "           Inheritance mode            Associated disease(s)  \n",
       "Locus                                                         \n",
       "PRNP     Autosomal dominant        Creutzfeldt-Jakob disease  \n",
       "VWA1    Autosomal recessive      Hereditary motor neuropathy  \n",
       "EIF4A3  Autosomal recessive  Richieri-Costa-Pereira syndrome  \n",
       "NIPA1    Autosomal dominant    Amyotrophic lateral sclerosis  "
      ]
     },
     "execution_count": 27,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "gnom.loc[patho[patho['source'] == 'gnomad'].index]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 28,
   "id": "49e9042c-b631-4b02-b154-d59108ea538b",
   "metadata": {},
   "outputs": [],
   "source": [
    "n_patho = patho.join(gnom[\"Inheritance mode\"])"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 30,
   "id": "d3f89d89-4352-43ef-9838-c3ffb44c206c",
   "metadata": {},
   "outputs": [],
   "source": [
    "n_patho.to_csv(\"Patho_v2.txt\", sep='\\t', index=False)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 4,
   "id": "a5cd8fbe-db05-4217-a0c2-d58ce4c5b803",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Locus</th>\n",
       "      <th>Disease</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>AFF2</td>\n",
       "      <td>FRAXE mental retardation (FRAXE)</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>AFF3</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>AR</td>\n",
       "      <td>Spinal and bulbar muscular atrophy (SBMA)</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>ARX_1</td>\n",
       "      <td>Developmental and epileptic encephalopathy-1 (...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>ARX_2</td>\n",
       "      <td>Developmental and epileptic encephalopathy-1 (...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>61</th>\n",
       "      <td>ZNF713</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>62</th>\n",
       "      <td>PRNP</td>\n",
       "      <td>Creutzfeldt-Jakob disease</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>63</th>\n",
       "      <td>VWA1</td>\n",
       "      <td>Hereditary motor neuropathy</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>64</th>\n",
       "      <td>EIF4A3</td>\n",
       "      <td>Richieri-Costa-Pereira syndrome</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>65</th>\n",
       "      <td>NIPA1</td>\n",
       "      <td>Amyotrophic lateral sclerosis</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>66 rows × 2 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "     Locus                                            Disease\n",
       "0     AFF2                   FRAXE mental retardation (FRAXE)\n",
       "1     AFF3                                                NaN\n",
       "2       AR          Spinal and bulbar muscular atrophy (SBMA)\n",
       "3    ARX_1  Developmental and epileptic encephalopathy-1 (...\n",
       "4    ARX_2  Developmental and epileptic encephalopathy-1 (...\n",
       "..     ...                                                ...\n",
       "61  ZNF713                                                NaN\n",
       "62    PRNP                          Creutzfeldt-Jakob disease\n",
       "63    VWA1                        Hereditary motor neuropathy\n",
       "64  EIF4A3                    Richieri-Costa-Pereira syndrome\n",
       "65   NIPA1                      Amyotrophic lateral sclerosis\n",
       "\n",
       "[66 rows x 2 columns]"
      ]
     },
     "execution_count": 4,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "data = pd.read_csv(\"Patho.tsv\", sep='\\t')\n",
    "data[[\"Locus\", \"Disease\"]]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "c2f58fe9-3202-4ec0-b8d5-e6d469dd06da",
   "metadata": {},
   "outputs": [],
   "source": []
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "Python 3 (ipykernel)",
   "language": "python",
   "name": "python3"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.9.6"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 5
}
