# Versions:
## v0.2 - Useable version
(Click the badge to go to the download page)  
[![DOI](https://zenodo.org/badge/DOI/10.5281/zenodo.7013709.svg)](https://doi.org/10.5281/zenodo.7013709)

Much simpler and much smaller data

### CHANGES:
* Using a pVCF created from 172 haplotype-resolved long-read assemblies, we removed any TR region which had no observed
  non-SNP variant. This removed 58% of the regions.
* With fewer regions, we simplified the data's structure and now store everything in a single tab-delimited bed-like
  file with columns:

### File Structure:
Columns:
* chrom - chromosome of TR region
* start - 0-based start position of TR region
* end - 0-based end position of TR region
* annos - Json containing a list of TRF annotated repeats with structure:
  * chrom - chromosome
  * start - start position of the repeat
  * end - end position of the repeat
  * period - period size of the repeat
  * copies - number of copes of the repeat in the reference
  * score - alignment score
  * entropy - entropy measure based on percent composition
  * repeat - motif of the repeat

Example:
```
chr11   11605859        11605958        [{"chrom": "chr11", "start": 11605912, "end": 11605927, "period": 2.0, "copies": 8.5, "score": 41, "entropy": 0.99, "repeat": "CA"}, {"chrom": "chr11", "start": 11605930, "end": 11605941, "period": 6.0, "copies": 2.0, "score": 36, "entropy": 1.92, "repeat": "AGCTTC"}]
```
### Notes
The single file of regions paired with their annotations allows much easier parsing/usage.
A custom parser can easily be built by splitting each line on tabs and using a json parser on the 4th column.
A parser has already been built into [truvari](https://github.com/ACEnglish/truvari) and can be used via:

```python
from truvari.annotations.trf import iter_tr_regions
# generator for every region
adotto_all_regions = iter_tr_regions("adotto_TRannotations_v0.2.bed.gz")
# fetch regions in a region
adotto_fetch_regions = iter_tr_regions("adotto_TRannotations_v0.2.bed.gz", region=("chr17", 10350000, 10360000))
```

Additionally, the file can be queried with [tabix](http://www.htslib.org/doc/tabix.html)
```bash
tabix adotto_TRannotations_v0.2.bed.gz chr11:11600000-11606000
```

## v0.1 - Initial version
(Click the badge to go to the download page)  
[![DOI](https://zenodo.org/badge/DOI/10.5281/zenodo.6930202.svg)](https://doi.org/10.5281/zenodo.6930202)

Inside of data_list.txt is the relative paths to data generated by this step.

We bundle these files using tar such that they can be recovered in-place into a copy of the code for reanalysis and to
keep things organized.

The tarball can be created with the command:

```bash
cat data_list.txt | tar czvf adotto_regions_data_<version>.tgz -T -
```

The tarball can be placed inside this directory and extracted in-place via:

```bash
tar xzvf adotto_regions_data_<version>.tgz
```
Note! the data_list.txt is put into .gitignore to help keep `git status` clean 

### CHANGES:
* It exists

### Notes:
List of files and their descriptions:
* `data/tr_regions.bed.gz` - Final set of tandem-repeat regions for analysis
* `data/tr_annotated.bed.gz` - TandemRepeatFinder annotations over the tandem-repeat regions
* `data/unannotated_regions.bed.gz` - tr_regions.bed.gz which have no accompanying tr_annotated.bed.gz entries
* `data/merged.slop25.bed.gz` - Merged calls from the sources with 25bp of slop added to each end
* `data/giab/giab_concat_input.bed.gz` - Raw input provided by GIAB from [ftp](https://ftp-trace.ncbi.nlm.nih.gov/ReferenceSamples/giab/release/genome-stratifications/v3.0/GRCh38/LowComplexity/)
* `data/giab/merged.bed.gz` - GIAB regions merged
* `data/baylor/grch38.simpleRepeat.truvari.bed.gz` - Raw input provided by baylor from [UCSC Simple repeat annotations](https://genome.ucsc.edu/cgi-bin/hgTables?db=mm10&hgta_group=varRep&hgta_track=simpleRepeat&hgta_table=simpleRepeat&hgta_doSchema=describe+table+schema)
* `data/baylor/merged.bed.gz` - Baylor regions merged
* `data/pacbio/repeat_catalog.hg38.bed.gz` - Raw input provided by pacbio from [Illumina](https://github.com/illumina/Repeatcatalogs)
* `data/pacbio/merged.bed.gz` - Pacbio regions merged
* `data/ucsd1/ensembleTR_loci_list.bed.gz` - Raw input provided by UCSD from ensemble
* `data/ucsd1/merged.bed.gz` - UCSD1 regions merged
* `data/ucsd2/GIAB_adVNTR_short_VNTR_regions.bed.gz` - Raw input provided by UCSD from ???
* `data/ucsd2/merged.bed.gz` - UCSD2 regions merged
